// ng-uikit/esm2015/lib/carousel/carousel.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, EventEmitter, Input, Output } from '@angular/core';
export class CarouselComponent {
    constructor() {
        this.items = [];
        this.afterNext = new EventEmitter();
        this.afterPrev = new EventEmitter();
        this.width = '600px';
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        this.activeIndex = 0;
        this.count = this.items.length;
    }
    /**
     * @return {?}
     */
    next() {
        (this.activeIndex < this.count - 1) ? this.activeIndex++ : this.activeIndex = 0;
        this.afterNext.emit({
            index: this.activeIndex,
            item: this.items[this.activeIndex]
        });
    }
    /**
     * @return {?}
     */
    prev() {
        (this.activeIndex >= 1) ? this.activeIndex-- : this.activeIndex = this.count - 1;
        this.afterPrev.emit({
            index: this.activeIndex,
            item: this.items[this.activeIndex]
        });
    }
    /**
     * @param {?} i
     * @return {?}
     */
    goTo(i) {
        this.activeIndex = i;
    }
}
CarouselComponent.decorators = [
    { type: Component, args: [{
                selector: 'ui-carousel',
                template: `<div id="carouselExampleIndicators" class="carousel slide" data-ride="carousel" [ngStyle]="{width: width}">
  <ol class="carousel-indicators">
      <li data-target="#carouselExampleIndicators" attr.data-slide-to="{{i}}" *ngFor="let item of items; let i=index;"
          [ngClass]="{'active':i == activeIndex}" (click)="goTo(i)">
      </li>
  </ol>
  <div class="carousel-inner">
    <div class="carousel-item" *ngFor="let item of items; let i=index;" [ngClass]="{'active':i == activeIndex}">
      <img src="{{item.src}}" alt="{{item.alt}}">
      <div class="carousel-caption d-none d-md-block">
        <h5>{{item.alt}}</h5>
        <p>{{item.alt}}</p>
      </div>
    </div>
  </div>
  <a class="carousel-control-prev" (click)="prev()" role="button" data-slide="prev">
    <span class="carousel-control-prev-icon" aria-hidden="true"></span>
    <span class="sr-only">Previous</span>
  </a>
  <a class="carousel-control-next" (click)="next()" role="button" data-slide="next">
    <span class="carousel-control-next-icon" aria-hidden="true"></span>
    <span class="sr-only">Next</span>
  </a>
</div>
`,
                styles: [``]
            },] },
];
CarouselComponent.ctorParameters = () => [];
CarouselComponent.propDecorators = {
    items: [{ type: Input }],
    afterNext: [{ type: Output }],
    afterPrev: [{ type: Output }]
};
if (false) {
    /** @type {?} */
    CarouselComponent.prototype.items;
    /** @type {?} */
    CarouselComponent.prototype.afterNext;
    /** @type {?} */
    CarouselComponent.prototype.afterPrev;
    /** @type {?} */
    CarouselComponent.prototype.activeIndex;
    /** @type {?} */
    CarouselComponent.prototype.count;
    /** @type {?} */
    CarouselComponent.prototype.width;
}
