// ng-uikit/bundles/ng-uikit.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/common'), require('@angular/common/http'), require('@angular/forms'), require('@angular/platform-browser')) :
    typeof define === 'function' && define.amd ? define('ng-uikit', ['exports', '@angular/core', '@angular/common', '@angular/common/http', '@angular/forms', '@angular/platform-browser'], factory) :
    (factory((global['ng-uikit'] = {}),global.ng.core,global.ng.common,global.ng.common.http,global.ng.forms,global.ng.platformBrowser));
}(this, (function (exports,i0,common,http,forms,platformBrowser) { 'use strict';

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var NgUikitComponent = (function () {
        function NgUikitComponent() {
        }
        /**
         * @return {?}
         */
        NgUikitComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
            };
        NgUikitComponent.decorators = [
            { type: i0.Component, args: [{
                        selector: 'lib-ng-uikit',
                        template: "\n    <p>\n      ng-uikit works!\n    </p>\n  ",
                        styles: []
                    },] },
        ];
        NgUikitComponent.ctorParameters = function () { return []; };
        return NgUikitComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var NgUikitModule = (function () {
        function NgUikitModule() {
        }
        NgUikitModule.decorators = [
            { type: i0.NgModule, args: [{
                        imports: [],
                        declarations: [NgUikitComponent],
                    },] },
        ];
        return NgUikitModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var AccordionComponent = (function () {
        function AccordionComponent() {
        }
        /**
         * @return {?}
         */
        AccordionComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
            };
        AccordionComponent.decorators = [
            { type: i0.Component, args: [{
                        selector: 'ui-accordion',
                        template: "<div class=\"accordion\" id=\"accordionExample\">\n  <ng-content></ng-content>\n</div>\n",
                        styles: [""]
                    },] },
        ];
        AccordionComponent.ctorParameters = function () { return []; };
        return AccordionComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var AccordionPanelHeaderComponent = (function () {
        function AccordionPanelHeaderComponent() {
        }
        /**
         * @return {?}
         */
        AccordionPanelHeaderComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
            };
        AccordionPanelHeaderComponent.decorators = [
            { type: i0.Component, args: [{
                        selector: 'ui-accordion-panel-header',
                        template: "<ng-content></ng-content>\n",
                        styles: [""]
                    },] },
        ];
        AccordionPanelHeaderComponent.ctorParameters = function () { return []; };
        return AccordionPanelHeaderComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var AccordionPanelComponent = (function () {
        function AccordionPanelComponent() {
        }
        Object.defineProperty(AccordionPanelComponent.prototype, "hasHeader", {
            get: /**
             * @return {?}
             */ function () {
                return this.headerComponent && this.headerComponent.length > 0;
            },
            enumerable: true,
            configurable: true
        });
        /**
         * @return {?}
         */
        AccordionPanelComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
                this.index = Math.ceil(1000 * Math.random());
            };
        /**
         * @param {?} i
         * @return {?}
         */
        AccordionPanelComponent.prototype.toggle = /**
         * @param {?} i
         * @return {?}
         */
            function (i) {
                this.isOpen = !this.isOpen;
            };
        AccordionPanelComponent.decorators = [
            { type: i0.Component, args: [{
                        selector: 'ui-accordion-panel',
                        template: "<div class=\"card\">\n  <div class=\"card-header\" [attr.id]=\"'heading'+index\">\n    <h5 class=\"mb-0\">\n      <a     data-toggle=\"collapse\" [attr.data-target]=\"'#collapse'+index\"\n              [attr.aria-expanded]=\"isOpen\" [attr.aria-controls]=\"'collapse'+index\" (click)=\"toggle(index)\">\n        <span class=\"ui-accordion-header-text\" *ngIf=\"!hasHeader\">\n          {{header}}\n        </span>\n        <ng-content select=\"ui-accordion-panel-header\" *ngIf=\"hasHeader\"></ng-content>\n      </a>\n    </h5>\n  </div>\n  <div [attr.id]=\"'collapse'+index\" class=\"collapse \" [ngClass]=\"{'show':isOpen}\"\n       [attr.aria-labelledby]=\"'heading'+index\" data-parent=\"#accordionExample\">\n    <div class=\"card-body\">\n      <ng-content></ng-content>\n    </div>\n  </div>\n</div>\n",
                        styles: [""]
                    },] },
        ];
        AccordionPanelComponent.ctorParameters = function () { return []; };
        AccordionPanelComponent.propDecorators = {
            header: [{ type: i0.Input }],
            isOpen: [{ type: i0.Input }],
            headerComponent: [{ type: i0.ContentChildren, args: [AccordionPanelHeaderComponent,] }]
        };
        return AccordionPanelComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var AccordionModule = (function () {
        function AccordionModule() {
        }
        AccordionModule.decorators = [
            { type: i0.NgModule, args: [{
                        imports: [
                            common.CommonModule
                        ],
                        declarations: [AccordionComponent, AccordionPanelComponent, AccordionPanelHeaderComponent],
                        exports: [AccordionComponent, AccordionPanelComponent, AccordionPanelHeaderComponent]
                    },] },
        ];
        return AccordionModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var AutocompleteComponent = (function () {
        function AutocompleteComponent(http$$1) {
            this.http = http$$1;
            this.label = 'Name';
            this.selected = new i0.EventEmitter();
            this.helpText = 'Enter at least 2 char to search';
            this.filteredList = [];
            this.selectedItem = (this.niddle) ? {} : "";
        }
        /**
         * @return {?}
         */
        AutocompleteComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
                if (this.sourceURL) {
                    this.loadResponse();
                }
            };
        /**
         * @param {?} key
         * @return {?}
         */
        AutocompleteComponent.prototype.filter = /**
         * @param {?} key
         * @return {?}
         */
            function (key) {
                if (key && key.length >= 2) {
                    this.filteredList = this.filterResult(key);
                    console.log(this.filteredList);
                }
                else {
                    this.filteredList = [];
                    this.selectedItem = (this.niddle) ? {} : "";
                }
            };
        /**
         * @param {?} key
         * @return {?}
         */
        AutocompleteComponent.prototype.filterResult = /**
         * @param {?} key
         * @return {?}
         */
            function (key) {
                var _this = this;
                /** @type {?} */
                var result = [];
                this.source.forEach(function (item) {
                    /** @type {?} */
                    var name = (_this.niddle) ? item[_this.niddle].toLowerCase() : item.toString().toLowerCase();
                    if (name.indexOf(key.toLowerCase()) !== -1 && (result.length < 10)) {
                        result.push(item);
                    }
                    else {
                        return true;
                    }
                });
                return result;
            };
        /**
         * @return {?}
         */
        AutocompleteComponent.prototype.loadResponse = /**
         * @return {?}
         */
            function () {
                var _this = this;
                /** @type {?} */
                var url = this.sourceURL || '../../assets/autocomplete.json';
                this.http.get(url).subscribe(function (res) {
                    if (Array.isArray(res)) {
                        _this.source = res;
                    }
                });
            };
        /**
         * @param {?} item
         * @return {?}
         */
        AutocompleteComponent.prototype.getText = /**
         * @param {?} item
         * @return {?}
         */
            function (item) {
                return (this.niddle) ? item[this.niddle] : item;
            };
        /**
         * @param {?} item
         * @return {?}
         */
        AutocompleteComponent.prototype.selectItem = /**
         * @param {?} item
         * @return {?}
         */
            function (item) {
                this.selectedItem = item;
                this.filteredList = [];
                this.selected.emit(item);
            };
        AutocompleteComponent.decorators = [
            { type: i0.Component, args: [{
                        selector: 'ui-autocomplete',
                        template: "<div class=\"form-group ui-autocomplete\">\n  <label for=\"UIAutocomplete\">{{label}}</label>\n  <input *ngIf=\"niddle\" type=\"text\" class=\"form-control\" id=\"UIAutocomplete1\" aria-describedby=\"emailHelp\" placeholder=\"Enter {{label}}\"\n         [ngModel]=\"selectedItem[niddle]\" (keyup)=\"filter($event.target.value)\" autocomplete=\"off\">\n\n  <input *ngIf=\"!niddle\" type=\"text\" class=\"form-control\" id=\"UIAutocomplete\" aria-describedby=\"emailHelp\" placeholder=\"Enter {{label}}\"\n         [ngModel]=\"selectedItem\" (keyup)=\"filter($event.target.value)\" autocomplete=\"off\">\n  <small id=\"emailHelp\" class=\"form-text text-muted\" *ngIf=\"helpText\">{{helpText}}</small>\n  <div class=\"ui-autocomplete-container\">\n    <ul class=\"list-group ui-autocomplete-list\" *ngIf=\"filteredList.length\">\n      <li class=\"list-group-item ui-autocomplete-list-item\" (click)=\"selectItem(item)\" *ngFor=\"let item of filteredList\">{{getText(item)}}</li>\n    </ul>\n  </div>\n</div>\n",
                        styles: [".ui-autocomplete-container{position:relative}.ui-autocomplete-list{position:absolute;z-index:9;width:100%}.ui-autocomplete-list-item:hover{cursor:pointer;color:#fff;background-color:#007bff;border-color:#007bff}"]
                    },] },
        ];
        AutocompleteComponent.ctorParameters = function () {
            return [
                { type: http.HttpClient }
            ];
        };
        AutocompleteComponent.propDecorators = {
            source: [{ type: i0.Input }],
            sourceURL: [{ type: i0.Input }],
            niddle: [{ type: i0.Input }],
            label: [{ type: i0.Input }],
            selected: [{ type: i0.Output }],
            helpText: [{ type: i0.Input }]
        };
        return AutocompleteComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var AutocompleteModule = (function () {
        function AutocompleteModule() {
        }
        AutocompleteModule.decorators = [
            { type: i0.NgModule, args: [{
                        imports: [
                            common.CommonModule,
                            forms.FormsModule
                        ],
                        declarations: [AutocompleteComponent],
                        exports: [AutocompleteComponent]
                    },] },
        ];
        return AutocompleteModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var CardComponent = (function () {
        function CardComponent() {
        }
        /**
         * @return {?}
         */
        CardComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
            };
        CardComponent.decorators = [
            { type: i0.Component, args: [{
                        selector: 'ui-card',
                        template: "<div class=\"card\" style=\"width: 18rem;\" *ngIf=\"config\">\n  <img class=\"card-img-top\" src=\"{{config.src}}\" alt=\"{{config.alt || config.title}}\" *ngIf=\"config.src\">\n  <div class=\"card-body\">\n    <h5 class=\"card-title\">{{config.title}}</h5>\n    <ng-content></ng-content>\n  </div>\n</div>\n",
                        styles: [""]
                    },] },
        ];
        CardComponent.ctorParameters = function () { return []; };
        CardComponent.propDecorators = {
            config: [{ type: i0.Input }]
        };
        return CardComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var CardModule = (function () {
        function CardModule() {
        }
        CardModule.decorators = [
            { type: i0.NgModule, args: [{
                        imports: [
                            common.CommonModule
                        ],
                        declarations: [CardComponent],
                        exports: [CardComponent]
                    },] },
        ];
        return CardModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var CarouselComponent = (function () {
        function CarouselComponent() {
            this.items = [];
            this.afterNext = new i0.EventEmitter();
            this.afterPrev = new i0.EventEmitter();
            this.width = '600px';
        }
        /**
         * @return {?}
         */
        CarouselComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
                this.activeIndex = 0;
                this.count = this.items.length;
            };
        /**
         * @return {?}
         */
        CarouselComponent.prototype.next = /**
         * @return {?}
         */
            function () {
                (this.activeIndex < this.count - 1) ? this.activeIndex++ : this.activeIndex = 0;
                this.afterNext.emit({
                    index: this.activeIndex,
                    item: this.items[this.activeIndex]
                });
            };
        /**
         * @return {?}
         */
        CarouselComponent.prototype.prev = /**
         * @return {?}
         */
            function () {
                (this.activeIndex >= 1) ? this.activeIndex-- : this.activeIndex = this.count - 1;
                this.afterPrev.emit({
                    index: this.activeIndex,
                    item: this.items[this.activeIndex]
                });
            };
        /**
         * @param {?} i
         * @return {?}
         */
        CarouselComponent.prototype.goTo = /**
         * @param {?} i
         * @return {?}
         */
            function (i) {
                this.activeIndex = i;
            };
        CarouselComponent.decorators = [
            { type: i0.Component, args: [{
                        selector: 'ui-carousel',
                        template: "<div id=\"carouselExampleIndicators\" class=\"carousel slide\" data-ride=\"carousel\" [ngStyle]=\"{width: width}\">\n  <ol class=\"carousel-indicators\">\n      <li data-target=\"#carouselExampleIndicators\" attr.data-slide-to=\"{{i}}\" *ngFor=\"let item of items; let i=index;\"\n          [ngClass]=\"{'active':i == activeIndex}\" (click)=\"goTo(i)\">\n      </li>\n  </ol>\n  <div class=\"carousel-inner\">\n    <div class=\"carousel-item\" *ngFor=\"let item of items; let i=index;\" [ngClass]=\"{'active':i == activeIndex}\">\n      <img src=\"{{item.src}}\" alt=\"{{item.alt}}\">\n      <div class=\"carousel-caption d-none d-md-block\">\n        <h5>{{item.alt}}</h5>\n        <p>{{item.alt}}</p>\n      </div>\n    </div>\n  </div>\n  <a class=\"carousel-control-prev\" (click)=\"prev()\" role=\"button\" data-slide=\"prev\">\n    <span class=\"carousel-control-prev-icon\" aria-hidden=\"true\"></span>\n    <span class=\"sr-only\">Previous</span>\n  </a>\n  <a class=\"carousel-control-next\" (click)=\"next()\" role=\"button\" data-slide=\"next\">\n    <span class=\"carousel-control-next-icon\" aria-hidden=\"true\"></span>\n    <span class=\"sr-only\">Next</span>\n  </a>\n</div>\n",
                        styles: [""]
                    },] },
        ];
        CarouselComponent.ctorParameters = function () { return []; };
        CarouselComponent.propDecorators = {
            items: [{ type: i0.Input }],
            afterNext: [{ type: i0.Output }],
            afterPrev: [{ type: i0.Output }]
        };
        return CarouselComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var CarouselModule = (function () {
        function CarouselModule() {
        }
        CarouselModule.decorators = [
            { type: i0.NgModule, args: [{
                        imports: [
                            common.CommonModule,
                            platformBrowser.BrowserModule
                        ],
                        declarations: [CarouselComponent],
                        exports: [CarouselComponent]
                    },] },
        ];
        return CarouselModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var ChartComponent = (function () {
        function ChartComponent(el) {
            this.el = el;
            this.width = '400px';
            this.height = '400px';
            this.clickChart = new i0.EventEmitter();
            this.responsive = true;
        }
        /**
         * @return {?}
         */
        ChartComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
            };
        /**
         * @return {?}
         */
        ChartComponent.prototype.ngAfterViewInit = /**
         * @return {?}
         */
            function () {
                /** @type {?} */
                var ctx = this.el.nativeElement.children[0].children[0].getContext('2d');
                /** @type {?} */
                var _that = this;
                Chart.plugins.register({
                    afterDatasetsDraw: function (chartInstance, easing) {
                        // To only draw at the end of animation, check for easing === 1
                        /** @type {?} */
                        var ctx = chartInstance.chart.ctx;
                        chartInstance.data.datasets.forEach(function (dataset, i) {
                            /** @type {?} */
                            var meta = chartInstance.getDatasetMeta(i);
                            if (!meta.hidden) {
                                meta.data.forEach(function (element, index) {
                                    // Draw the text in black, with the specified font
                                    ctx.fillStyle = 'rgb(0, 0, 0)';
                                    /** @type {?} */
                                    var fontSize = 16;
                                    /** @type {?} */
                                    var fontStyle = 'normal';
                                    /** @type {?} */
                                    var fontFamily = 'Helvetica Neue';
                                    ctx.font = Chart.helpers.fontString(fontSize, fontStyle, fontFamily);
                                    // Just naively convert to string for now
                                    /** @type {?} */
                                    var dataString = dataset.data[index].toString();
                                    // Make sure alignment settings are correct
                                    ctx.textAlign = 'center';
                                    ctx.textBaseline = 'middle';
                                    /** @type {?} */
                                    var padding = 5;
                                    /** @type {?} */
                                    var position = element.tooltipPosition();
                                    ctx.fillText(dataString, position.x, position.y - (fontSize / 2) - padding);
                                });
                            }
                        });
                    }
                });
                this.chart = new Chart(ctx, {
                    // The type of chart we want to create
                    type: 'bar',
                    // The data for our dataset
                    data: this.data,
                    maintainAspectRatio: false,
                    responsive: true,
                    // Configuration options go here
                    options: {
                        onClick: function (e) {
                            /** @type {?} */
                            var activePoints = _that.chart.getElementsAtEvent(e);
                            /** @type {?} */
                            var index = activePoints[0]._index;
                            _that.clickChart.emit({ y: this.data.datasets[0].data[index], x: this.data.labels[index] });
                        },
                        title: {
                            display: true,
                            text: 'My Title',
                            fontSize: 16
                        },
                        legend: {
                            enabled: false,
                        },
                        scales: {
                            yAxes: [{
                                    ticks: {
                                        beginAtZero: true
                                    }
                                }]
                        },
                        layout: {
                            padding: {
                                left: 50,
                                right: 0,
                                top: 0,
                                bottom: 0
                            }
                        }
                    },
                    onClick: function (e) {
                        /** @type {?} */
                        var activePoints = _that.chart.getElementsAtEvent(e);
                        /** @type {?} */
                        var selectedIndex = activePoints[0]._index;
                        console.log(this.data.datasets[0].data[selectedIndex]);
                    }
                });
            };
        /**
         * @param {?} e
         * @return {?}
         */
        ChartComponent.prototype.onCanvasClick = /**
         * @param {?} e
         * @return {?}
         */
            function (e) {
                // if(this.chart) {
                //   let element = this.chart.getElementAtEvent(event);
                //   let dataset = this.chart.getDatasetAtEvent(event);
                //   if(element && element[0] && dataset) {
                //     this.clickChart.emit({originalEvent: event, element: element[0], dataset: dataset});
                //   }
                // }
            };
        ChartComponent.decorators = [
            { type: i0.Component, args: [{
                        selector: 'ui-chart',
                        template: "<div style=\"position:relative\" [style.width]=\"responsive && !width ? null : width\" [style.height]=\"responsive && !height ? null : height\">\n  <canvas [attr.width]=\"responsive && !width ? null : width\" [attr.height]=\"responsive && !height ? null : height\" (click)=\"onCanvasClick($event)\"></canvas>\n</div>\n",
                        styles: [""]
                    },] },
        ];
        ChartComponent.ctorParameters = function () {
            return [
                { type: i0.ElementRef }
            ];
        };
        ChartComponent.propDecorators = {
            data: [{ type: i0.Input }],
            width: [{ type: i0.Input }],
            height: [{ type: i0.Input }],
            clickChart: [{ type: i0.Output }]
        };
        return ChartComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var ChartModule = (function () {
        function ChartModule() {
        }
        ChartModule.decorators = [
            { type: i0.NgModule, args: [{
                        imports: [
                            common.CommonModule
                        ],
                        declarations: [ChartComponent],
                        exports: [ChartComponent]
                    },] },
        ];
        return ChartModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var DatatableService = (function () {
        function DatatableService() {
        }
        /**
         * @param {?} rows
         * @param {?} size
         * @param {?} pageNum
         * @return {?}
         */
        DatatableService.prototype.getPageData = /**
         * @param {?} rows
         * @param {?} size
         * @param {?} pageNum
         * @return {?}
         */
            function (rows, size, pageNum) {
                /** @type {?} */
                var start = (pageNum - 1) * size;
                /** @type {?} */
                var last = (start + size) > rows.length ? rows.length : start + size;
                return {
                    startIndex: start,
                    rows: rows.slice(start, last),
                    lastIndex: last
                };
            };
        DatatableService.decorators = [
            { type: i0.Injectable, args: [{
                        providedIn: 'root'
                    },] },
        ];
        DatatableService.ctorParameters = function () { return []; };
        /** @nocollapse */ DatatableService.ngInjectableDef = i0.defineInjectable({ factory: function DatatableService_Factory() { return new DatatableService(); }, token: DatatableService, providedIn: "root" });
        return DatatableService;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var DatatableComponent = (function () {
        function DatatableComponent(_dts) {
            this._dts = _dts;
            this.pageChange = new i0.EventEmitter();
            this.sorted = new i0.EventEmitter();
            this.size = 6;
            this.search = {};
        }
        /**
         * @return {?}
         */
        DatatableComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
                this.page = this._dts.getPageData(this.rows, this.size, 1);
                console.log(this.page);
                this.resultCount = this.rows.length;
            };
        /**
         * @param {?} col
         * @return {?}
         */
        DatatableComponent.prototype.getColumnClass = /**
         * @param {?} col
         * @return {?}
         */
            function (col) {
                /** @type {?} */
                var className = ' ui-dt-cell ';
                if (col.className) {
                    className += col.className + ' ';
                }
                if (col.sortable) {
                    className += ' sortable';
                }
                return className;
            };
        /**
         * @param {?} e
         * @return {?}
         */
        DatatableComponent.prototype.onPageChange = /**
         * @param {?} e
         * @return {?}
         */
            function (e) {
                this.page = this._dts.getPageData(this.rows, this.size, e.pageNum);
            };
        /**
         * @param {?} item
         * @return {?}
         */
        DatatableComponent.prototype.sort = /**
         * @param {?} item
         * @return {?}
         */
            function (item) {
                this.page.rows.sort(this.dynamicSort(item.field));
            };
        /**
         * @param {?} property
         * @return {?}
         */
        DatatableComponent.prototype.dynamicSort = /**
         * @param {?} property
         * @return {?}
         */
            function (property) {
                /** @type {?} */
                var sortOrder = 1;
                if (property[0] === '-') {
                    sortOrder = -1;
                    property = property.substr(1);
                }
                return function (a, b) {
                    if (sortOrder === -1) {
                        return b[property].toString().localeCompare(a[property]);
                    }
                    else {
                        return a[property].toString().localeCompare(b[property]);
                    }
                };
            };
        /**
         * @param {?} event
         * @param {?} field
         * @param {?} val
         * @return {?}
         */
        DatatableComponent.prototype.filter = /**
         * @param {?} event
         * @param {?} field
         * @param {?} val
         * @return {?}
         */
            function (event, field, val) {
                if (val.toString().length) {
                    if (event.keyCode === 13 || event.which === 13) {
                        this.search = {};
                        this.search[field] = val;
                        console.log(field, this.search, this.rows);
                        /** @type {?} */
                        var rows_1 = [];
                        this.rows.forEach(function (item) {
                            if (item[field].toString().toLowerCase().indexOf(val.toString().toLowerCase()) !== -1) {
                                rows_1.push(item);
                            }
                        });
                        this.resultCount = rows_1.length;
                        this.page = this._dts.getPageData(rows_1, this.size, 1);
                    }
                }
                else {
                    this.resultCount = this.rows.length;
                    this.page = this._dts.getPageData(this.rows, this.size, 1);
                }
            };
        DatatableComponent.decorators = [
            { type: i0.Component, args: [{
                        selector: 'ui-datatable',
                        template: "<div class=\"ui-dt-table\">\n  <div class=\"ui-dt-thead row\">\n    <div *ngFor=\"let col of columns\" [ngClass]=\"getColumnClass(col)\"  >\n      {{col.label}}\n      <i class=\"fas fa-sort-amount-up\" (click)=\"sort(col)\"></i>\n    </div>\n  </div>\n  <div class=\"ui-dt-thead row\">\n    <div *ngFor=\"let col of columns\" [ngClass]=\"getColumnClass(col)\"  >\n        <input type=\"text\" class=\"form-control form-control-sm\"\n               [name]=\"col.field\"\n               [ngModel]=\"search[col.field]\"\n               (keyup)=\"filter($event, col.field, $event.target.value)\"\n        >\n    </div>\n  </div>\n  <div class=\"ui-dt-tbody\">\n    <div *ngFor=\"let row of page.rows;let i=index;\" class=\"row ui-dt-row\">\n      <div [attr.class]=\"col.className +' ui-dt-cell'\" *ngFor=\"let col of columns;let k=index;\">\n         {{row[col.field]}}\n      </div>\n    </div>\n  </div>\n</div>\n<ui-pagination\n  [size]=\"size\"\n  [totalCount]=\"resultCount\"\n  (pageChange)=\"onPageChange($event)\">\n</ui-pagination>\n\n",
                        styles: [".ui-dt-thead{border-bottom:1px solid #999}.ui-dt-thead .ui-dt-cell{font-weight:500}.ui-dt-row{border-bottom:1px solid #999}.ui-dt-cell{-ms-word-break:break-word;word-break:break-word;border-left:1px solid #ddd;padding:5px}.ui-dt-row .ui-dt-cell:last-child{border-right:1px solid #ddd}"]
                    },] },
        ];
        DatatableComponent.ctorParameters = function () {
            return [
                { type: DatatableService }
            ];
        };
        DatatableComponent.propDecorators = {
            rows: [{ type: i0.Input }],
            columns: [{ type: i0.Input }],
            pageChange: [{ type: i0.Output }],
            sorted: [{ type: i0.Output }],
            size: [{ type: i0.Input }]
        };
        return DatatableComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var PagingService = (function () {
        function PagingService() {
        }
        Object.defineProperty(PagingService.prototype, "size", {
            get: /**
             * @return {?}
             */ function () {
                return this._size;
            },
            set: /**
             * @param {?} value
             * @return {?}
             */ function (value) {
                this._size = value;
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(PagingService.prototype, "totalCount", {
            get: /**
             * @return {?}
             */ function () {
                return this._totalCount;
            },
            set: /**
             * @param {?} value
             * @return {?}
             */ function (value) {
                this._totalCount = value;
            },
            enumerable: true,
            configurable: true
        });
        /**
         * @param {?} totalCount
         * @return {?}
         */
        PagingService.prototype.getListOfPage = /**
         * @param {?} totalCount
         * @return {?}
         */
            function (totalCount) {
                return Math.ceil(totalCount / this._size);
            };
        PagingService.decorators = [
            { type: i0.Injectable, args: [{
                        providedIn: 'root'
                    },] },
        ];
        PagingService.ctorParameters = function () { return []; };
        /** @nocollapse */ PagingService.ngInjectableDef = i0.defineInjectable({ factory: function PagingService_Factory() { return new PagingService(); }, token: PagingService, providedIn: "root" });
        return PagingService;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var PaginationComponent = (function () {
        function PaginationComponent(_pgs) {
            this._pgs = _pgs;
            this.size = 10;
            this.pageChange = new i0.EventEmitter();
            this.pageNum = 1;
            this.displayNumbers = 5;
            this.lastPage = 5;
        }
        /**
         * @return {?}
         */
        PaginationComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
                this._pgs.size = this.size;
                this._pgs.totalCount = this.totalCount;
                this.pageCount = this._pgs.getListOfPage(this.totalCount);
                this.totalPages = Array(this.pageCount).fill(0).map(function (e, i) { return i + 1; });
                this.pages = this.totalPages.slice(0, 5);
                this.lastPage = this.pages[this.displayNumbers - 1];
            };
        /**
         * @param {?} e
         * @return {?}
         */
        PaginationComponent.prototype.onPageChange = /**
         * @param {?} e
         * @return {?}
         */
            function (e) {
                console.log(e);
                this.pageNum = e;
                this.pageChange.emit({
                    pageNum: e
                });
            };
        /**
         * @param {?=} isReverse
         * @return {?}
         */
        PaginationComponent.prototype.shiftBoundary = /**
         * @param {?=} isReverse
         * @return {?}
         */
            function (isReverse) {
                if (isReverse === void 0) {
                    isReverse = false;
                }
                /** @type {?} */
                var startIndex = (isReverse) ? this.pageNum - this.displayNumbers : this.pageNum - 1;
                /** @type {?} */
                var lastIndex = (isReverse) ? this.pageNum : this.pageNum + this.displayNumbers - 1;
                this.pages = this.totalPages.slice(startIndex, lastIndex);
                this.lastPage = lastIndex;
            };
        /**
         * @return {?}
         */
        PaginationComponent.prototype.next = /**
         * @return {?}
         */
            function () {
                this.pageNum++;
                if (this.pageNum > this.lastPage)
                    this.shiftBoundary();
            };
        /**
         * @return {?}
         */
        PaginationComponent.prototype.prev = /**
         * @return {?}
         */
            function () {
                this.pageNum--;
                if ((this.pageNum) % this.displayNumbers === 0) {
                    this.shiftBoundary(true);
                }
            };
        PaginationComponent.decorators = [
            { type: i0.Component, args: [{
                        selector: 'ui-pagination',
                        template: "<div class=\"m-3\">\n  <nav aria-label=\"Page navigation example\">\n    <ul class=\"pagination justify-content-center\">\n      <ng-container>\n\n        <li class=\"page-item \" [ngClass]=\"{'disabled':pageNum === 1}\" (click)=\"prev()\" [attr.disabled]=\"pageNum === 1\">\n          <a class=\"page-link\" tabindex=\"-1\">Previous </a>\n        </li>\n\n        <li class=\"page-item\" (click)=\"onPageChange(page)\" *ngFor=\"let page of pages; index as i; first as isFirst; last as isLast;\">\n          <a class=\"page-link\" [ngClass]=\"{'active disabled':pageNum===page}\">{{page}}</a>\n        </li>\n\n        <li class=\"page-item\" *ngIf=\" pageCount > displayNumbers && lastPage < pageCount\" [ngClass]=\"{'disabled':pageNum===pages.length}\" (click)=\"shiftBoundary()\">\n          <a class=\"page-link\">...</a>\n        </li>\n\n        <li class=\"page-item\" *ngIf=\"pageCount > displayNumbers && lastPage < pageCount\" [ngClass]=\"{'disabled':pageNum===pageCount}\" (click)=\"onPageChange(pageCount)\">\n          <a class=\"page-link\"> {{pageCount}}</a>\n        </li>\n\n        <li class=\"page-item\" [ngClass]=\"{'disabled':pageNum === pageCount}\" [attr.disabled]=\"(pageNum === pageCount)?'disabled':''\" (click)=\"next()\">\n          <a class=\"page-link\"> Next</a>\n        </li>\n      </ng-container>\n    </ul>\n  </nav>\n</div>\n\n",
                        styles: [".page-link:not([href]):not([tabindex]){position:relative;display:block;padding:.5rem .75rem;margin-left:-1px;line-height:1.25;color:#337ab7;background-color:#fff;border:1px solid #dee2e6}.page-link.active:not([href]):not([tabindex]){position:relative;display:block;padding:.5rem .75rem;margin-left:-1px;line-height:1.25;color:#fff;background-color:#337ab7;border:1px solid #337ab7}.disabled{pointer-events:none}"]
                    },] },
        ];
        PaginationComponent.ctorParameters = function () {
            return [
                { type: PagingService }
            ];
        };
        PaginationComponent.propDecorators = {
            totalCount: [{ type: i0.Input }],
            size: [{ type: i0.Input }],
            pageChange: [{ type: i0.Output }]
        };
        return PaginationComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var DatatableModule = (function () {
        function DatatableModule() {
        }
        DatatableModule.decorators = [
            { type: i0.NgModule, args: [{
                        imports: [
                            common.CommonModule,
                            forms.FormsModule
                        ],
                        providers: [DatatableService],
                        declarations: [DatatableComponent, PaginationComponent],
                        exports: [DatatableComponent, PaginationComponent]
                    },] },
        ];
        return DatatableModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    /** @type {?} */
    var DefaultLocaleMonths = 'January_February_March_April_May_June_July_August_September_October_November_December'.split('_');
    /** @type {?} */
    var DefaultLocaleMonthsShort = 'Jan_Feb_Mar_Apr_May_Jun_Jul_Aug_Sep_Oct_Nov_Dec'.split('_');
    /** @type {?} */
    var DefaultLocaleWeekdays = 'Sunday_Monday_Tuesday_Wednesday_Thursday_Friday_Saturday'.split('_');
    /** @type {?} */
    var DefaultLocaleWeekdaysShort = 'Sun_Mon_Tue_Wed_Thu_Fri_Sat'.split('_');

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var DatepickerService = (function () {
        function DatepickerService() {
        }
        Object.defineProperty(DatepickerService.prototype, "dateObj", {
            get: /**
             * @return {?}
             */ function () {
                return this._dateObj;
            },
            set: /**
             * @param {?} value
             * @return {?}
             */ function (value) {
                this._dateObj = value;
            },
            enumerable: true,
            configurable: true
        });
        /**
         * @return {?}
         */
        DatepickerService.prototype.getDatesOfMonth = /**
         * @return {?}
         */
            function () {
                /** @type {?} */
                var d = this.dateObj;
                /** @type {?} */
                var y = d.getFullYear();
                /** @type {?} */
                var m = d.getMonth();
                /** @type {?} */
                var f = new Date(y, m, 1);
                /** @type {?} */
                var l = new Date(y, m + 1, 0);
                /** @type {?} */
                var days = [
                    [null, null, null, null, null, null, null],
                    [null, null, null, null, null, null, null],
                    [null, null, null, null, null, null, null],
                    [null, null, null, null, null, null, null],
                    [null, null, null, null, null, null, null],
                    [null, null, null, null, null, null, null]
                ];
                console.log(l.getDate());
                /** @type {?} */
                var c = f.getDay();
                /** @type {?} */
                var row = 0;
                for (var i = 1; i <= l.getDate(); i++) {
                    days[row][c] = (i <= 9) ? '0' + i : i + '';
                    c++;
                    if (c === 7) {
                        row++;
                        c = 0;
                    }
                }
                console.log(row, days);
                return days;
            };
        /**
         * @param {?} year
         * @param {?} month
         * @param {?} day
         * @param {?} hours
         * @param {?} minutes
         * @param {?} seconds
         * @return {?}
         */
        DatepickerService.prototype.getDateJsonObj = /**
         * @param {?} year
         * @param {?} month
         * @param {?} day
         * @param {?} hours
         * @param {?} minutes
         * @param {?} seconds
         * @return {?}
         */
            function (year, month, day, hours, minutes, seconds) {
                this.dateObj = new Date(year, month, day, hours, minutes, seconds);
                /** @type {?} */
                var d = this.dateObj;
                /** @type {?} */
                var date = (d.getDate() <= 9) ? '0' + d.getDate() : d.getDate() + '';
                /** @type {?} */
                var monthNumeric = (d.getMonth() <= 9) ? '0' + d.getMonth() : d.getMonth() + '';
                return {
                    day: d.getDay(),
                    dayString: DefaultLocaleWeekdays[d.getDay()],
                    dayStringShort: DefaultLocaleWeekdaysShort[d.getDay()],
                    date: date,
                    month: monthNumeric,
                    monthString: DefaultLocaleMonths[d.getUTCMonth()],
                    monthStringShort: DefaultLocaleMonthsShort[d.getUTCMonth()],
                    time: d.getTime().toString(),
                    hours: d.getHours().toString(),
                    minutes: d.getMinutes().toString(),
                    seconds: d.getSeconds().toString(),
                    year: d.getFullYear().toString(),
                    UTCString: d.toUTCString(),
                    numericDate: d.getFullYear() + '-' + monthNumeric + '-' + date + '-' + d.getHours() + '-' + d.getSeconds(),
                    formatDate: this.getFormatDate(d)
                };
            };
        /**
         * @param {?} format
         * @return {?}
         */
        DatepickerService.prototype.setFormat = /**
         * @param {?} format
         * @return {?}
         */
            function (format) {
                this.format = format;
            };
        /**
         * @private
         * @param {?} d
         * @return {?}
         */
        DatepickerService.prototype.getFormatDate = /**
         * @private
         * @param {?} d
         * @return {?}
         */
            function (d) {
                /** @type {?} */
                var dateStr = this.format;
                console.log(d.getFullYear());
                dateStr = dateStr.replace('yyyy', d.getFullYear());
                dateStr = dateStr.replace('mm', this.getMonthString(d));
                dateStr = dateStr.replace('dd', d.getDate());
                console.log(dateStr);
                return dateStr;
            };
        /**
         * @param {?} date
         * @return {?}
         */
        DatepickerService.prototype.getMonthString = /**
         * @param {?} date
         * @return {?}
         */
            function (date) {
                console.log(date.getMonth());
                /** @type {?} */
                var month = date.getMonth() + 1;
                return month < 10 ? '0' + month : '' + month; // ('' + month) for string result
            };
        DatepickerService.decorators = [
            { type: i0.Injectable, args: [{
                        providedIn: 'root'
                    },] },
        ];
        DatepickerService.ctorParameters = function () { return []; };
        /** @nocollapse */ DatepickerService.ngInjectableDef = i0.defineInjectable({ factory: function DatepickerService_Factory() { return new DatepickerService(); }, token: DatepickerService, providedIn: "root" });
        return DatepickerService;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var DatepickerComponent = (function () {
        function DatepickerComponent(datepickerSvc) {
            this.datepickerSvc = datepickerSvc;
            this.selected = new i0.EventEmitter();
            this.mode = 'single';
            this.items = [
                {
                    url: 'a',
                    text: 'Action',
                    isDivider: true
                }, {
                    url: 'a',
                    text: 'Another action'
                }, {
                    url: 'a',
                    text: 'Something else here'
                }
            ];
            this.defaultLocaleWeekdaysShort = DefaultLocaleWeekdaysShort;
            this.range = {
                from: null,
                to: null
            };
            this.showCal = false;
        }
        /**
         * @return {?}
         */
        DatepickerComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
                this.labelText = 'Select Date';
                /** @type {?} */
                var d = new Date();
                this.datepickerSvc.setFormat(this.format || 'yyyy-mm-dd');
                this.selectedDateTime = this.datepickerSvc.getDateJsonObj(d.getFullYear(), d.getMonth(), d.getDate(), d.getHours(), d.getMinutes(), d.getSeconds());
                this.daysList = this.datepickerSvc.getDatesOfMonth();
            };
        /**
         * @param {?} day
         * @return {?}
         */
        DatepickerComponent.prototype.isWithinRange = /**
         * @param {?} day
         * @return {?}
         */
            function (day) {
                if (day) {
                    /** @type {?} */
                    var a = this.selectedDateTime.numericDate.split('-');
                    a[2] = (day <= 9) ? '0' + day : day + '';
                    /** @type {?} */
                    var today = parseInt(a.join(''), 10);
                    /** @type {?} */
                    var from = parseInt(this.range.from.numericDate.split('-').join(''), 10);
                    /** @type {?} */
                    var to = parseInt(this.range.to.numericDate.split('-').join(''), 10);
                    return (today >= from && today <= to);
                }
                return false;
            };
        /**
         * @param {?} day
         * @return {?}
         */
        DatepickerComponent.prototype.getSelectedClass = /**
         * @param {?} day
         * @return {?}
         */
            function (day) {
                /** @type {?} */
                var isSelected = false;
                day = parseInt(day, 10);
                if (this.mode === 'range') {
                    if (this.range.from) {
                        isSelected = (this.range.from.date === day);
                    }
                    if (this.range.from && this.range.to) {
                        isSelected = this.isWithinRange(day);
                    }
                }
                else {
                    isSelected = this.selectedDateTime.date === day;
                }
                return { 'c-datepicker__day--selected': isSelected };
            };
        /**
         * @param {?} day
         * @return {?}
         */
        DatepickerComponent.prototype.setDate = /**
         * @param {?} day
         * @return {?}
         */
            function (day) {
                /** @type {?} */
                var d = this.selectedDateTime;
                this.selectedDateTime = this.datepickerSvc.getDateJsonObj(d.year, d.month, day, d.hours, d.minutes, d.seconds);
                if (this.mode === 'range') {
                    if (this.range.from) {
                        this.range.to = this.selectedDateTime;
                        this.selectedDate = this.range.from.formatDate + " - " + this.range.to.formatDate;
                    }
                    else {
                        this.range.from = this.selectedDateTime;
                        this.selectedDate = this.range.from.formatDate + " ";
                    }
                    this.selected.emit(this.range);
                }
                else {
                    this.selectedDate = this.selectedDateTime.formatDate;
                    this.showCal = false;
                    this.selected.emit(this.selectedDateTime);
                }
            };
        /**
         * @return {?}
         */
        DatepickerComponent.prototype.prevMonth = /**
         * @return {?}
         */
            function () {
                /** @type {?} */
                var d = this.selectedDateTime;
                this.selectedDateTime = this.datepickerSvc.getDateJsonObj(d.year, parseInt(d.month, 10) - 1, d.date, d.hours, d.minutes, d.seconds);
                this.daysList = this.datepickerSvc.getDatesOfMonth();
                this.selected.emit(this.selectedDateTime);
            };
        /**
         * @return {?}
         */
        DatepickerComponent.prototype.nextMonth = /**
         * @return {?}
         */
            function () {
                /** @type {?} */
                var d = this.selectedDateTime;
                this.selectedDateTime = this.datepickerSvc.getDateJsonObj(d.year, parseInt(d.month, 10) + 1, d.date, d.hours, d.minutes, d.seconds);
                this.daysList = this.datepickerSvc.getDatesOfMonth();
                this.selected.emit(this.selectedDateTime);
            };
        /**
         * @return {?}
         */
        DatepickerComponent.prototype.clear = /**
         * @return {?}
         */
            function () {
                this.range = {
                    from: null,
                    to: null
                };
                this.selectedDate = '';
            };
        DatepickerComponent.decorators = [
            { type: i0.Component, args: [{
                        selector: 'ui-datepicker',
                        template: "<div class=\"form-group\">\n  <label for=\"basic-url\">{{labelText}}</label>\n  <div class=\"input-group mb-3\">\n    <div class=\"input-group-prepend\">\n      <span class=\"input-group-text\">\n        <i class=\"far fa-calendar-alt\"></i>\n      </span>\n    </div>\n    <input type=\"text\" class=\"form-control\" placeholder=\"Date\" (focus)=\"showCal=true\" aria-label=\"Date\"\n           aria-describedby=\"basic-addon1\" [ngModel]=\"selectedDate\" id=\"basic-url\">\n  </div>\n</div>\n<div class=\"datepicker-container\">\n  <div class=\"c-datepicker c-datepicker--open\" *ngIf=\"showCal\">\n    <a class=\"c-datepicker__toggle c-datepicker__toggle--right c-datepicker--show-time js-show-clock\"\n       title=\"show time picker\">\n      <i class=\"far fa-clock\"></i>\n    </a>\n\n    <div\n      class=\"c-datepicker__toggle c-datepicker__toggle--left c-datepicker--show-calendar js-show-calendar is-selected\">\n      <!--&nbsp;\n      <a class=\"\"\n         title=\"show date picker\">\n        <i class=\"far fa-calendar-alt\"></i>\n      </a>\n      &nbsp;-->\n\n      <i class=\"far fa-trash-alt\" title=\"clear datepicker\" (click)=\"clear()\"></i>\n\n    </div>\n\n\n    <div class=\"c-datepicker__header\">\n      <div class=\"c-datepicker__header-day\">\n        <span class=\"js-day\">{{selectedDateTime.formatDate}}</span>\n      </div>\n      <!--<div class=\"c-datepicker__header-date\">-->\n      <!--<span class=\"c-datepicker__header-date__month js-date-month\">Jun 2019</span>-->\n      <!--<span class=\"c-datepicker__header-date__day js-date-day\">10th</span>-->\n      <!--<span class=\"c-datepicker__header-date__time js-date-time\">-->\n      <!--<span class=\"c-datepicker__header-date__hours js-date-hours active\">12</span>:<span-->\n      <!--class=\"c-datepicker__header-date__minutes js-date-minutes\">00</span>-->\n      <!--</span>-->\n      <!--</div>-->\n    </div>\n\n    <div class=\"c-datepicker__calendar\" data-rome-id=\"0\">\n      <div class=\"c-datepicker__calendar\" style=\"display: inline-block;\">\n        <div class=\"c-datepicker__date\">\n          <div class=\"c-datepicker__month\">\n            <button class=\"c-datepicker__back\" type=\"button\" (click)=\"prevMonth()\"><i class=\"fas fa-chevron-left\"></i>\n            </button>\n            <button class=\"c-datepicker__next\" type=\"button\" (click)=\"nextMonth()\"><i class=\"fas fa-chevron-right\"></i>\n            </button>\n            <div class=\"rd-month-label\">{{selectedDateTime.monthStringShort}} {{selectedDateTime.year}}</div>\n            <table class=\"c-datepicker__days\">\n              <thead class=\"c-datepicker__days-head\">\n              <!-- Days Sun - Sat -->\n              <tr class=\"c-datepicker__days-row\">\n                <th class=\"c-datepicker__day-head\" *ngFor=\"let day  of defaultLocaleWeekdaysShort\">{{day}}</th>\n              </tr>\n              </thead>\n              <tbody class=\"c-datepicker__days-body\" data-rome-offset=\"0\">\n              <!-- Date 1-31 -->\n              <tr class=\"c-datepicker__days-row\" *ngFor=\"let row of daysList\">\n                <td class=\"c-datepicker__day-body\" *ngFor=\"let day of row\" [ngClass]=\"getSelectedClass(day)\"\n                    (click)=\"setDate(day)\">\n                  <span>{{day}}</span>\n                </td>\n              </tr>\n              <!--<tr class=\"c-datepicker__days-row\">-->\n              <!--<td class=\"c-datepicker__day-body rd-day-prev-month\">26</td>-->\n              <!--<td class=\"c-datepicker__day-body c-datepicker__day selected\">10</td>-->\n              <!--<td class=\"c-datepicker__day-body rd-day-next-month\">01</td>-->\n              <!--</tr>  -->\n              </tbody>\n            </table>\n          </div>\n        </div>\n      </div>\n    </div>\n    <div class=\"c-datepicker__clock\">\n      <div class=\"c-datepicker__clock__am-pm-toggle\">\n        <label class=\"\">\n          <input class=\"c-datepicker__toggle c-datepicker__toggle--right c-datepicker__clock--am\" type=\"radio\"\n                 name=\"time-date-toggle\" value=\"AM\">\n          AM\n        </label>\n\n        <label class=\"c-datepicker__toggle--checked\">\n          <input class=\"c-datepicker__toggle c-datepicker__toggle--right c-datepicker__clock--pm\" type=\"radio\"\n                 name=\"time-date-toggle\" value=\"PM\" checked=\"checked\">\n          PM\n        </label>\n      </div>\n      <div class=\"c-datepicker__mask\"></div>\n      <div class=\"c-datepicker__clock__hours js-clock-hours active\">\n        <div class=\"c-datepicker__clock__num\" data-number=\"3\">3</div>\n        <div class=\"c-datepicker__clock__num\" data-number=\"4\">4</div>\n        <div class=\"c-datepicker__clock__num\" data-number=\"5\">5</div>\n        <div class=\"c-datepicker__clock__num\" data-number=\"6\">6</div>\n        <div class=\"c-datepicker__clock__num\" data-number=\"7\">7</div>\n        <div class=\"c-datepicker__clock__num\" data-number=\"8\">8</div>\n        <div class=\"c-datepicker__clock__num\" data-number=\"9\">9</div>\n        <div class=\"c-datepicker__clock__num\" data-number=\"10\">10</div>\n        <div class=\"c-datepicker__clock__num\" data-number=\"11\">11</div>\n        <div class=\"c-datepicker__clock__num c-datepicker__clock__num--active\" data-number=\"0\">12</div>\n        <div class=\"c-datepicker__clock__num\" data-number=\"1\">1</div>\n        <div class=\"c-datepicker__clock__num\" data-number=\"2\">2</div>\n        <div class=\"c-datepicker__clock-hands\">\n          <div class=\"c-datepicker__hour-hand\"></div>\n        </div>\n      </div>\n      <div class=\"c-datepicker__clock__minutes js-clock-minutes\">\n        <div class=\"c-datepicker__clock__num\" data-number=\"15\">15</div>\n        <div class=\"c-datepicker__clock__num\" data-number=\"20\">20</div>\n        <div class=\"c-datepicker__clock__num\" data-number=\"25\">25</div>\n        <div class=\"c-datepicker__clock__num\" data-number=\"30\">30</div>\n        <div class=\"c-datepicker__clock__num\" data-number=\"35\">35</div>\n        <div class=\"c-datepicker__clock__num\" data-number=\"40\">40</div>\n        <div class=\"c-datepicker__clock__num\" data-number=\"45\">45</div>\n        <div class=\"c-datepicker__clock__num\" data-number=\"50\">50</div>\n        <div class=\"c-datepicker__clock__num\" data-number=\"55\">55</div>\n        <div class=\"c-datepicker__clock__num c-datepicker__clock__num--active\" data-number=\"0\">0</div>\n        <div class=\"c-datepicker__clock__num\" data-number=\"5\">5</div>\n        <div class=\"c-datepicker__clock__num\" data-number=\"10\">10</div>\n        <div class=\"c-datepicker__clock-hands\">\n          <div class=\"c-datepicker__hour-hand\"></div>\n        </div>\n      </div>\n    </div>\n    <div class=\"modal-btns\">\n      <a class=\"c-btn c-btn--flat js-cancel\" (click)=\"clear()\">Cancel</a>\n      <a class=\"c-btn c-btn--flat js-ok\" (click)=\"showCal=false\">OK</a>\n    </div>\n  </div>\n</div>\n",
                        styles: [".datepicker-container{position:relative;z-index:999}.c-scrim{position:fixed;left:0;right:0;top:0;bottom:0;background-color:rgba(0,0,0,.541176);opacity:0;transition:opacity .2s;will-change:opacity}.c-scrim--shown{opacity:1}.c-datepicker{min-height:390px;position:absolute;background:#fff;border:0;width:300px;text-align:center;-webkit-tap-highlight-color:transparent;box-shadow:0 14px 45px rgba(0,0,0,.25),0 10px 18px rgba(0,0,0,.22);border-radius:2px;opacity:0;will-change:opacity;transition:opacity .2s ease-in-out,top .2s ease-in-out}.c-datepicker--open{opacity:1;top:50%}.c-datepicker__header{position:relative}.c-datepicker__header-day{height:32px;background:#0097a7;color:#fff;line-height:32px;font-size:12px;font-weight:200;letter-spacing:.3px}.c-datepicker__header::after{content:\"\";display:table;clear:both}.c-datepicker__header-date{background:#00bcd4;height:150px;padding:16px 0}.rd-month-label{height:56px;line-height:56px;font-size:14px;font-weight:800}.c-datepicker__back,.c-datepicker__next,.c-datepicker__toggle{position:absolute;border:0;background:0 0;line-height:1;width:56px;height:56px;display:inline-block;overflow:hidden;-webkit-font-smoothing:antialiased;cursor:pointer}.c-datepicker__back:focus,.c-datepicker__next:focus,.c-datepicker__toggle:focus{outline:0}.c-datepicker__back{left:0}.c-datepicker__next{right:0}.c-datepicker--show-calendar:after,.c-datepicker--show-time:after{color:#fff;visibility:visible}.c-datepicker__header-date span{display:block;color:#fff;margin:0;transition:opacity .1s ease-in-out}.c-datepicker__header-date__month{cursor:pointer;font-size:24px;opacity:.6}.c-datepicker__header-date__day{cursor:pointer;font-size:64px;opacity:.6}.c-datepicker__header-date__time{font-size:25px;opacity:.6}.c-datepicker__header-date__time>span{display:inline-block}.c-datepicker__header-date__hours,.c-datepicker__header-date__minutes{cursor:pointer}.c-datepicker--show-time.is-selected~.c-datepicker__header .c-datepicker__header-date__time{opacity:1}.c-datepicker--show-time.is-selected~.c-datepicker__header .c-datepicker__header-date__time .c-datepicker__header-date__hours,.c-datepicker--show-time.is-selected~.c-datepicker__header .c-datepicker__header-date__time .c-datepicker__header-date__minutes{opacity:.6}.c-datepicker--show-calendar.is-selected~.c-datepicker__header .c-datepicker__header-date__day,.c-datepicker--show-calendar.is-selected~.c-datepicker__header .c-datepicker__header-date__month,.c-datepicker--show-time.is-selected~.c-datepicker__header .c-datepicker__header-date__time .c-datepicker__header-date__hours.active,.c-datepicker--show-time.is-selected~.c-datepicker__header .c-datepicker__header-date__time .c-datepicker__header-date__minutes.active{opacity:1}.modal-btns{padding:20px;position:absolute;bottom:0;right:0;float:right}.c-datepicker__day-body{font-size:12px;color:rgba(0,0,0,.8);width:36px;height:36px}.c-datepicker__day--selected::after{content:\"\";position:absolute;left:50%;top:50%;width:35px;height:35px;border-radius:50%;-webkit-transform:translate(-50%,-50%);transform:translate(-50%,-50%);background:rgba(0,0,0,.05)}.c-datepicker__day-head{color:rgba(0,0,0,.54);font-size:12px;height:36px}.c-datepicker__day-head,c-datepicker__day-body{-webkit-tap-highlight-color:transparent}.c-btn{display:inline-block;min-width:56px;cursor:pointer}.rd-day-next-month,.rd-day-prev-month{opacity:.1;pointer-events:none}.c-datepicker__calendar{height:300px}.c-datepicker__date{position:absolute;left:0;right:0}.c-datepicker__days{margin:10px 20px}.c-datepicker__header-toggle{position:absolute;top:50%;color:#fff;cursor:pointer}.c-datepicker__header-toggle i{font-size:26px}.c-datepicker__header-toggle--left{left:20px}.c-datepicker__header-toggle--right{right:20px}.c-datepicker__header-toggle--inactive{opacity:.2}.c-datepicker__toggle{top:7px;width:45px;height:30px;color:#fff!important;z-index:1;transition:opacity .2s ease-in-out}.c-datepicker__toggle--right{right:0}.c-datepicker__toggle--left{left:0}.c-datepicker__toggle.is-selected{opacity:1}.c-datepicker--show-calendar.is-selected~.c-datepicker__clock,.c-datepicker--show-time.is-selected~.c-datepicker__calendar{display:none}.c-datepicker__clock{position:relative;width:200px;height:200px;padding:160px 0 20px;border-radius:50%;list-style:none;font-size:14px;line-height:50px;margin:0 auto}.c-datepicker__clock .c-datepicker__clock__num{display:block;position:absolute;top:50%;left:50%;width:50px;height:50px;margin:-25px;z-index:98}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(1){-webkit-transform:rotate(0) translate(100px) rotate(0);transform:rotate(0) translate(100px) rotate(0)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(1).c-datepicker__clock__num--active:not(.hide-hand)~.c-datepicker__clock-hands{-webkit-transform:translate(-50%,-50%) rotate(270deg);transform:translate(-50%,-50%) rotate(270deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(1).c-datepicker__clock__num--active:not(.hide-hand)~.c-datepicker__clock-hands .c-datepicker__hour-hand{opacity:1;background:#00bcd4}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(1):hover~.c-datepicker__clock-hands{-webkit-transform:translate(-50%,-50%) rotate(270deg);transform:translate(-50%,-50%) rotate(270deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(1):hover~.c-datepicker__clock-hands .c-datepicker__hour-hand{opacity:1;background:#00bcd4}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(2){-webkit-transform:rotate(30deg) translate(100px) rotate(-30deg);transform:rotate(30deg) translate(100px) rotate(-30deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(2).c-datepicker__clock__num--active:not(.hide-hand)~.c-datepicker__clock-hands{-webkit-transform:translate(-50%,-50%) rotate(300deg);transform:translate(-50%,-50%) rotate(300deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(2).c-datepicker__clock__num--active:not(.hide-hand)~.c-datepicker__clock-hands .c-datepicker__hour-hand{opacity:1;background:#00bcd4}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(2):hover~.c-datepicker__clock-hands{-webkit-transform:translate(-50%,-50%) rotate(300deg);transform:translate(-50%,-50%) rotate(300deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(2):hover~.c-datepicker__clock-hands .c-datepicker__hour-hand{opacity:1;background:#00bcd4}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(3){-webkit-transform:rotate(60deg) translate(100px) rotate(-60deg);transform:rotate(60deg) translate(100px) rotate(-60deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(3).c-datepicker__clock__num--active:not(.hide-hand)~.c-datepicker__clock-hands{-webkit-transform:translate(-50%,-50%) rotate(330deg);transform:translate(-50%,-50%) rotate(330deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(3).c-datepicker__clock__num--active:not(.hide-hand)~.c-datepicker__clock-hands .c-datepicker__hour-hand{opacity:1;background:#00bcd4}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(3):hover~.c-datepicker__clock-hands{-webkit-transform:translate(-50%,-50%) rotate(330deg);transform:translate(-50%,-50%) rotate(330deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(3):hover~.c-datepicker__clock-hands .c-datepicker__hour-hand{opacity:1;background:#00bcd4}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(4){-webkit-transform:rotate(90deg) translate(100px) rotate(-90deg);transform:rotate(90deg) translate(100px) rotate(-90deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(4).c-datepicker__clock__num--active:not(.hide-hand)~.c-datepicker__clock-hands{-webkit-transform:translate(-50%,-50%) rotate(360deg);transform:translate(-50%,-50%) rotate(360deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(4).c-datepicker__clock__num--active:not(.hide-hand)~.c-datepicker__clock-hands .c-datepicker__hour-hand{opacity:1;background:#00bcd4}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(4):hover~.c-datepicker__clock-hands{-webkit-transform:translate(-50%,-50%) rotate(360deg);transform:translate(-50%,-50%) rotate(360deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(4):hover~.c-datepicker__clock-hands .c-datepicker__hour-hand{opacity:1;background:#00bcd4}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(5){-webkit-transform:rotate(120deg) translate(100px) rotate(-120deg);transform:rotate(120deg) translate(100px) rotate(-120deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(5).c-datepicker__clock__num--active:not(.hide-hand)~.c-datepicker__clock-hands{-webkit-transform:translate(-50%,-50%) rotate(390deg);transform:translate(-50%,-50%) rotate(390deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(5).c-datepicker__clock__num--active:not(.hide-hand)~.c-datepicker__clock-hands .c-datepicker__hour-hand{opacity:1;background:#00bcd4}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(5):hover~.c-datepicker__clock-hands{-webkit-transform:translate(-50%,-50%) rotate(390deg);transform:translate(-50%,-50%) rotate(390deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(5):hover~.c-datepicker__clock-hands .c-datepicker__hour-hand{opacity:1;background:#00bcd4}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(6){-webkit-transform:rotate(150deg) translate(100px) rotate(-150deg);transform:rotate(150deg) translate(100px) rotate(-150deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(6).c-datepicker__clock__num--active:not(.hide-hand)~.c-datepicker__clock-hands{-webkit-transform:translate(-50%,-50%) rotate(420deg);transform:translate(-50%,-50%) rotate(420deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(6).c-datepicker__clock__num--active:not(.hide-hand)~.c-datepicker__clock-hands .c-datepicker__hour-hand{opacity:1;background:#00bcd4}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(6):hover~.c-datepicker__clock-hands{-webkit-transform:translate(-50%,-50%) rotate(420deg);transform:translate(-50%,-50%) rotate(420deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(6):hover~.c-datepicker__clock-hands .c-datepicker__hour-hand{opacity:1;background:#00bcd4}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(7){-webkit-transform:rotate(180deg) translate(100px) rotate(-180deg);transform:rotate(180deg) translate(100px) rotate(-180deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(7).c-datepicker__clock__num--active:not(.hide-hand)~.c-datepicker__clock-hands{-webkit-transform:translate(-50%,-50%) rotate(450deg);transform:translate(-50%,-50%) rotate(450deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(7).c-datepicker__clock__num--active:not(.hide-hand)~.c-datepicker__clock-hands .c-datepicker__hour-hand{opacity:1;background:#00bcd4}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(7):hover~.c-datepicker__clock-hands{-webkit-transform:translate(-50%,-50%) rotate(450deg);transform:translate(-50%,-50%) rotate(450deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(7):hover~.c-datepicker__clock-hands .c-datepicker__hour-hand{opacity:1;background:#00bcd4}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(8){-webkit-transform:rotate(210deg) translate(100px) rotate(-210deg);transform:rotate(210deg) translate(100px) rotate(-210deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(8).c-datepicker__clock__num--active:not(.hide-hand)~.c-datepicker__clock-hands{-webkit-transform:translate(-50%,-50%) rotate(480deg);transform:translate(-50%,-50%) rotate(480deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(8).c-datepicker__clock__num--active:not(.hide-hand)~.c-datepicker__clock-hands .c-datepicker__hour-hand{opacity:1;background:#00bcd4}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(8):hover~.c-datepicker__clock-hands{-webkit-transform:translate(-50%,-50%) rotate(480deg);transform:translate(-50%,-50%) rotate(480deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(8):hover~.c-datepicker__clock-hands .c-datepicker__hour-hand{opacity:1;background:#00bcd4}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(9){-webkit-transform:rotate(240deg) translate(100px) rotate(-240deg);transform:rotate(240deg) translate(100px) rotate(-240deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(9).c-datepicker__clock__num--active:not(.hide-hand)~.c-datepicker__clock-hands{-webkit-transform:translate(-50%,-50%) rotate(510deg);transform:translate(-50%,-50%) rotate(510deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(9).c-datepicker__clock__num--active:not(.hide-hand)~.c-datepicker__clock-hands .c-datepicker__hour-hand{opacity:1;background:#00bcd4}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(9):hover~.c-datepicker__clock-hands{-webkit-transform:translate(-50%,-50%) rotate(510deg);transform:translate(-50%,-50%) rotate(510deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(9):hover~.c-datepicker__clock-hands .c-datepicker__hour-hand{opacity:1;background:#00bcd4}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(10){-webkit-transform:rotate(270deg) translate(100px) rotate(-270deg);transform:rotate(270deg) translate(100px) rotate(-270deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(10).c-datepicker__clock__num--active:not(.hide-hand)~.c-datepicker__clock-hands{-webkit-transform:translate(-50%,-50%) rotate(540deg);transform:translate(-50%,-50%) rotate(540deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(10).c-datepicker__clock__num--active:not(.hide-hand)~.c-datepicker__clock-hands .c-datepicker__hour-hand{opacity:1;background:#00bcd4}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(10):hover~.c-datepicker__clock-hands{-webkit-transform:translate(-50%,-50%) rotate(540deg);transform:translate(-50%,-50%) rotate(540deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(10):hover~.c-datepicker__clock-hands .c-datepicker__hour-hand{opacity:1;background:#00bcd4}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(11){-webkit-transform:rotate(300deg) translate(100px) rotate(-300deg);transform:rotate(300deg) translate(100px) rotate(-300deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(11).c-datepicker__clock__num--active:not(.hide-hand)~.c-datepicker__clock-hands{-webkit-transform:translate(-50%,-50%) rotate(570deg);transform:translate(-50%,-50%) rotate(570deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(11).c-datepicker__clock__num--active:not(.hide-hand)~.c-datepicker__clock-hands .c-datepicker__hour-hand{opacity:1;background:#00bcd4}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(11):hover~.c-datepicker__clock-hands{-webkit-transform:translate(-50%,-50%) rotate(570deg);transform:translate(-50%,-50%) rotate(570deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(11):hover~.c-datepicker__clock-hands .c-datepicker__hour-hand{opacity:1;background:#00bcd4}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(12){-webkit-transform:rotate(330deg) translate(100px) rotate(-330deg);transform:rotate(330deg) translate(100px) rotate(-330deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(12).c-datepicker__clock__num--active:not(.hide-hand)~.c-datepicker__clock-hands{-webkit-transform:translate(-50%,-50%) rotate(600deg);transform:translate(-50%,-50%) rotate(600deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(12).c-datepicker__clock__num--active:not(.hide-hand)~.c-datepicker__clock-hands .c-datepicker__hour-hand{opacity:1;background:#00bcd4}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(12):hover~.c-datepicker__clock-hands{-webkit-transform:translate(-50%,-50%) rotate(600deg);transform:translate(-50%,-50%) rotate(600deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(12):hover~.c-datepicker__clock-hands .c-datepicker__hour-hand{opacity:1;background:#00bcd4}.c-datepicker__clock::before{content:\"\";position:absolute;top:70px;left:-20px;width:240px;height:240px;background:rgba(0,0,0,.05);border-radius:50%}.c-datepicker__clock__am-pm-toggle label,.c-datepicker__clock__num,.c-datepicker__day-body,.u-hover-ball-effect{position:relative;cursor:pointer}.c-datepicker__clock__am-pm-toggle label:before,.c-datepicker__clock__num:before,.c-datepicker__day-body:before,.u-hover-ball-effect:before{content:\"\";position:absolute;left:50%;top:50%;width:0%;height:0%;border-radius:50%;-webkit-transform:translate(-50%,-50%);transform:translate(-50%,-50%);transition:width .1s ease-in-out,height .1s ease-in-out}.c-datepicker__clock__am-pm-toggle label:hover,.c-datepicker__clock__num:hover,.c-datepicker__day-body:hover,.u-hover-ball-effect:hover{color:#fff}.c-datepicker__clock__am-pm-toggle label:hover:before,.c-datepicker__clock__num:hover:before,.c-datepicker__day-body:hover:before,.u-hover-ball-effect:hover:before{background:#00bcd4;width:35px;height:35px;z-index:-1}.c-datepicker__clock__num--active:not(.hide-hand),.c-datepicker__day-body--active:not(.hide-hand){color:#fff}.c-datepicker__clock__num--active:not(.hide-hand):before,.c-datepicker__day-body--active:not(.hide-hand):before{background:#00bcd4;width:35px;height:35px;z-index:-1}.c-datepicker__clock-hands{position:absolute;left:50%;top:50%;-webkit-transform:translate(-50%,-50%) rotate(180deg);transform:translate(-50%,-50%) rotate(180deg);width:10px;height:10px;border-radius:50%;background:#0097a7}.c-datepicker__hour-hand{position:absolute;opacity:0;height:78px;width:2px;background:#00bcd4;left:4px;top:10px}.c-datepicker__clock__minutes{height:200px;margin:-69px 0 0;width:200px;display:none}.c-datepicker__clock__minutes.active{display:block}.c-datepicker__clock__hours{height:200px;margin:-69px 0 0;width:200px;display:none}.c-datepicker__clock__hours.active{display:block}.c-datepicker__mask{width:127px;height:132px;position:absolute;top:122px;left:37px;z-index:99}.c-datepicker__mask:after{content:' ';width:156px;height:70px;display:block;position:absolute;top:32px;left:0;margin-left:-13px}.c-datepicker__mask:before{content:' ';width:75px;height:158px;display:block;position:absolute;top:6px;left:28px;margin-top:-18px}.c-datepicker__clock--show-minutes .c-datepicker__clock__minutes{visibility:visible}.c-datepicker__clock--show-hours .c-datepicker__clock__minutes,.c-datepicker__clock--show-minutes .c-datepicker__clock__hours{visibility:hidden}.c-datepicker__clock--show-hours .c-datepicker__clock__hours{visibility:visible}.c-datepicker__clock__am-pm-toggle{position:absolute;top:0;left:10px;right:10px;height:40px;padding:20px;line-height:40px}.c-datepicker__clock__am-pm-toggle label{width:40px;position:absolute}.c-datepicker__clock__am-pm-toggle label:nth-child(1){left:0}.c-datepicker__clock__am-pm-toggle label:nth-child(2){right:0}.c-datepicker__clock__am-pm-toggle label.c-datepicker__toggle--checked::after{content:\"\";position:absolute;left:50%;top:50%;border-radius:50%;-webkit-transform:translate(-50%,-50%);transform:translate(-50%,-50%);width:36px;height:36px;z-index:-1;background:rgba(0,0,0,.05)}.input-group{position:relative;display:flex;flex-wrap:wrap;align-items:stretch;width:100%}.input-group>.custom-file,.input-group>.custom-select,.input-group>.form-control,.input-group>.form-control-plaintext{position:relative;flex:1 1 auto;width:1%;margin-bottom:0}.input-group>.custom-file+.custom-file,.input-group>.custom-file+.custom-select,.input-group>.custom-file+.form-control,.input-group>.custom-select+.custom-file,.input-group>.custom-select+.custom-select,.input-group>.custom-select+.form-control,.input-group>.form-control+.custom-file,.input-group>.form-control+.custom-select,.input-group>.form-control+.form-control,.input-group>.form-control-plaintext+.custom-file,.input-group>.form-control-plaintext+.custom-select,.input-group>.form-control-plaintext+.form-control{margin-left:-1px}.input-group>.custom-file .custom-file-input:focus~.custom-file-label,.input-group>.custom-select:focus,.input-group>.form-control:focus{z-index:3}.input-group>.custom-file .custom-file-input:focus{z-index:4}.input-group>.custom-select:not(:last-child),.input-group>.form-control:not(:last-child){border-top-right-radius:0;border-bottom-right-radius:0}.input-group>.custom-select:not(:first-child),.input-group>.form-control:not(:first-child){border-top-left-radius:0;border-bottom-left-radius:0}.input-group>.custom-file{display:flex;align-items:center}.input-group>.custom-file:not(:last-child) .custom-file-label,.input-group>.custom-file:not(:last-child) .custom-file-label::after{border-top-right-radius:0;border-bottom-right-radius:0}.input-group>.custom-file:not(:first-child) .custom-file-label{border-top-left-radius:0;border-bottom-left-radius:0}.input-group-append,.input-group-prepend{display:flex}.input-group-append .btn,.input-group-prepend .btn{position:relative;z-index:2}.input-group-append .btn:focus,.input-group-prepend .btn:focus{z-index:3}.input-group-append .btn+.btn,.input-group-append .btn+.input-group-text,.input-group-append .input-group-text+.btn,.input-group-append .input-group-text+.input-group-text,.input-group-prepend .btn+.btn,.input-group-prepend .btn+.input-group-text,.input-group-prepend .input-group-text+.btn,.input-group-prepend .input-group-text+.input-group-text{margin-left:-1px}.input-group-prepend{margin-right:-1px}.input-group-append{margin-left:-1px}.input-group-text{display:flex;align-items:center;padding:.375rem .75rem;margin-bottom:0;font-size:1rem;font-weight:400;line-height:1.5;color:#495057;text-align:center;white-space:nowrap;background-color:#e9ecef;border:1px solid #ced4da;border-radius:.25rem}.input-group-text input[type=checkbox],.input-group-text input[type=radio]{margin-top:0}.input-group-lg>.custom-select,.input-group-lg>.form-control:not(textarea){height:calc(2.875rem + 2px)}.input-group-lg>.custom-select,.input-group-lg>.form-control,.input-group-lg>.input-group-append>.btn,.input-group-lg>.input-group-append>.input-group-text,.input-group-lg>.input-group-prepend>.btn,.input-group-lg>.input-group-prepend>.input-group-text{padding:.5rem 1rem;font-size:1.25rem;line-height:1.5;border-radius:.3rem}.input-group-sm>.custom-select,.input-group-sm>.form-control:not(textarea){height:calc(1.8125rem + 2px)}.input-group-sm>.custom-select,.input-group-sm>.form-control,.input-group-sm>.input-group-append>.btn,.input-group-sm>.input-group-append>.input-group-text,.input-group-sm>.input-group-prepend>.btn,.input-group-sm>.input-group-prepend>.input-group-text{padding:.25rem .5rem;font-size:.875rem;line-height:1.5;border-radius:.2rem}.input-group-lg>.custom-select,.input-group-sm>.custom-select{padding-right:1.75rem}.input-group>.input-group-append:last-child>.btn:not(:last-child):not(.dropdown-toggle),.input-group>.input-group-append:last-child>.input-group-text:not(:last-child),.input-group>.input-group-append:not(:last-child)>.btn,.input-group>.input-group-append:not(:last-child)>.input-group-text,.input-group>.input-group-prepend>.btn,.input-group>.input-group-prepend>.input-group-text{border-top-right-radius:0;border-bottom-right-radius:0}.input-group>.input-group-append>.btn,.input-group>.input-group-append>.input-group-text,.input-group>.input-group-prepend:first-child>.btn:not(:first-child),.input-group>.input-group-prepend:first-child>.input-group-text:not(:first-child),.input-group>.input-group-prepend:not(:first-child)>.btn,.input-group>.input-group-prepend:not(:first-child)>.input-group-text{border-top-left-radius:0;border-bottom-left-radius:0}"]
                    },] },
        ];
        DatepickerComponent.ctorParameters = function () {
            return [
                { type: DatepickerService }
            ];
        };
        DatepickerComponent.propDecorators = {
            selected: [{ type: i0.Output }],
            mode: [{ type: i0.Input }],
            format: [{ type: i0.Input }],
            labelText: [{ type: i0.Input }]
        };
        return DatepickerComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var DropdownComponent = (function () {
        function DropdownComponent() {
            this.items = [];
            this.buttonText = 'Your Button Text';
            this.isShowDropdown = false;
        }
        /**
         * @return {?}
         */
        DropdownComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
                this.isShowDropdown = this.isOpen || false;
            };
        /**
         * @return {?}
         */
        DropdownComponent.prototype.toggle = /**
         * @return {?}
         */
            function () {
                this.isShowDropdown = !this.isShowDropdown;
            };
        DropdownComponent.decorators = [
            { type: i0.Component, args: [{
                        selector: 'ui-dropdown',
                        template: "<div class=\"dropdown show\" [ngClass]=\"{'show': isShowDropdown}\">\n  <a class=\"btn btn-secondary dropdown-toggle\" (click)=\"toggle()\" role=\"button\" id=\"dropdownMenuLink\" data-toggle=\"dropdown\"\n     aria-haspopup=\"true\" attr.aria-expanded=\"{{isShowDropdown}}\">\n    {{buttonText}}\n  </a>\n  <div class=\"dropdown-menu dropdown-menu-right\" [ngClass]=\"{'show': isShowDropdown}\" aria-labelledby=\"dropdownMenuLink\">\n    <ng-container *ngFor=\"let item of items\">\n      <a class=\"dropdown-item\" href=\"{{item.url}}\" >{{item.text}}</a>\n      <div class=\"dropdown-divider\" *ngIf=\"item.isDivider\"></div>\n    </ng-container>\n    <ng-content></ng-content>\n  </div>\n</div>\n",
                        styles: [""]
                    },] },
        ];
        DropdownComponent.ctorParameters = function () { return []; };
        DropdownComponent.propDecorators = {
            items: [{ type: i0.Input }],
            buttonText: [{ type: i0.Input }],
            isOpen: [{ type: i0.Input }]
        };
        return DropdownComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var DropdownModule = (function () {
        function DropdownModule() {
        }
        DropdownModule.decorators = [
            { type: i0.NgModule, args: [{
                        imports: [
                            common.CommonModule
                        ],
                        declarations: [DropdownComponent],
                        exports: [DropdownComponent]
                    },] },
        ];
        return DropdownModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var DatepickerModule = (function () {
        function DatepickerModule() {
        }
        DatepickerModule.decorators = [
            { type: i0.NgModule, args: [{
                        imports: [
                            common.CommonModule,
                            platformBrowser.BrowserModule,
                            forms.FormsModule,
                            DropdownModule
                        ],
                        declarations: [DatepickerComponent],
                        exports: [DatepickerComponent]
                    },] },
        ];
        return DatepickerModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var ModalComponent = (function () {
        function ModalComponent() {
        }
        /**
         * @return {?}
         */
        ModalComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
            };
        /**
         * @return {?}
         */
        ModalComponent.prototype.toggle = /**
         * @return {?}
         */
            function () {
                this.showModel = !this.showModel;
            };
        ModalComponent.decorators = [
            { type: i0.Component, args: [{
                        selector: 'ui-modal',
                        template: "<!-- Button trigger modal -->\n<button type=\"button\" class=\"btn btn-primary\" data-toggle=\"modal\" data-target=\"#exampleModal\" (click)=\"toggle()\">\n  Launch demo modal\n</button>\n\n<!-- Modal -->\n<div class=\"modal fade\" id=\"exampleModal\" tabindex=\"-1\" role=\"dialog\" aria-labelledby=\"exampleModalLabel\" aria-hidden=\"true\" [ngClass]=\"{'show':showModel}\">\n  <div class=\"modal-dialog\" role=\"document\">\n    <div class=\"modal-content\">\n      <div class=\"modal-header\">\n        <h5 class=\"modal-title\" id=\"exampleModalLabel\">Modal title</h5>\n        <button type=\"button\" class=\"close\" data-dismiss=\"modal\" aria-label=\"Close\" (click)=\"toggle()\">\n          <span aria-hidden=\"true\">&times;</span>\n        </button>\n      </div>\n      <div class=\"modal-body\">\n        <ng-content></ng-content>\n      </div>\n      <div class=\"modal-footer\">\n        <button type=\"button\" class=\"btn btn-secondary\" data-dismiss=\"modal\" (click)=\"toggle()\">Close</button>\n        <button type=\"button\" class=\"btn btn-primary\" (click)=\"toggle()\">Save changes</button>\n      </div>\n    </div>\n  </div>\n</div>\n<div class=\"modal-backdrop fade show\" *ngIf=\"showModel\"></div>\n",
                        styles: [".show{display:block;padding-right:16px}"]
                    },] },
        ];
        ModalComponent.ctorParameters = function () { return []; };
        return ModalComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var ModalModule = (function () {
        function ModalModule() {
        }
        ModalModule.decorators = [
            { type: i0.NgModule, args: [{
                        imports: [
                            common.CommonModule
                        ],
                        declarations: [ModalComponent],
                        exports: [ModalComponent]
                    },] },
        ];
        return ModalModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var TabPanelComponent = (function () {
        function TabPanelComponent() {
        }
        /**
         * @return {?}
         */
        TabPanelComponent.prototype.ngOnChanges = /**
         * @return {?}
         */
            function () {
                if (this.isOpen) {
                    console.log(this.isOpen);
                }
            };
        /**
         * @return {?}
         */
        TabPanelComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
                this.index = Math.ceil(1000 * Math.random());
            };
        TabPanelComponent.decorators = [
            { type: i0.Component, args: [{
                        selector: 'ui-tab-panel',
                        template: "<div class=\"tab-pane\" [attr.id]=\"'tab'+index\" [ngClass]=\"{'active show':isOpen}\" role=\"tabpanel\"\n     aria-labelledby=\"home-tab\">\n  <ng-content></ng-content>\n</div>\n",
                        styles: [".tab-pane{display:none}.tab-pane.active{display:block}"]
                    },] },
        ];
        TabPanelComponent.ctorParameters = function () { return []; };
        TabPanelComponent.propDecorators = {
            header: [{ type: i0.Input }],
            isOpen: [{ type: i0.Input }]
        };
        return TabPanelComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var TabComponent = (function () {
        function TabComponent() {
            this.optionTabAfter = 3;
            this.isShowDropdown = false;
        }
        /**
         * @return {?}
         */
        TabComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
            };
        /**
         * @return {?}
         */
        TabComponent.prototype.toggleDropdown = /**
         * @return {?}
         */
            function () {
                this.isShowDropdown = !this.isShowDropdown;
            };
        /**
         * @param {?} i
         * @return {?}
         */
        TabComponent.prototype.toggle = /**
         * @param {?} i
         * @return {?}
         */
            function (i) {
                this.tabPanels.forEach(function (tab) {
                    tab.isOpen = (tab.index === i);
                });
            };
        /**
         * @return {?}
         */
        TabComponent.prototype.ngAfterContentInit = /**
         * @return {?}
         */
            function () {
                var _this = this;
                this.tabs = [];
                this.optionTabs = [];
                /** @type {?} */
                var count = 0;
                this.tabPanels.map(function (tab) {
                    /** @type {?} */
                    var tabItem = {
                        headerText: tab.header,
                        index: tab.index,
                        isOpen: !!tab
                    };
                    count++;
                    _this.optionTabAfter = (_this.optionTabAfter >= _this.tabPanels.length) ? _this.tabPanels.length : _this.optionTabAfter;
                    if (count <= _this.optionTabAfter) {
                        _this.tabs.push(tabItem);
                    }
                    else {
                        _this.optionTabs.push(tabItem);
                    }
                });
                console.log(this.tabs, this.optionTabs);
            };
        TabComponent.decorators = [
            { type: i0.Component, args: [{
                        selector: 'ui-tab',
                        template: "<ul class=\"nav nav-tabs\" id=\"myTab\" role=\"tablist\">\n  <li class=\"nav-item\" *ngFor=\"let item of tabs;index as i\">\n    <a class=\"nav-link \"\n       [attr.id]=\"'tab'+item.index\"\n       data-toggle=\"tab\"\n       (click)=\"toggle(item.index)\"\n       role=\"tab\" aria-controls=\"home\"\n       [attr.aria-selected]=\"item.isOpen\"\n       [ngClass]=\"{'active':item.isOpen}\">\n      {{item.headerText}}\n    </a>\n  </li>\n  <li role=\"presentation\" class=\"dropdown\" *ngIf=\"optionTabs.length\" [ngClass]=\"{'show': isShowDropdown}\">\n    <a class=\"dropdown-toggle\" id=\"myTabDrop\" (click)=\"toggleDropdown()\" data-toggle=\"dropdown\" aria-controls=\"myTabDrop-contents\" attr.aria-expanded=\"{{isShowDropdown}}\">more\n      <span class=\"caret\"></span>\n    </a>\n    <ul class=\"dropdown-menu\" aria-labelledby=\"myTabDrop\" id=\"myTabDrop1-contents\" [ngClass]=\"{'show': isShowDropdown}\">\n      <li *ngFor=\"let item of optionTabs\">\n        <a [attr.id]=\"'tab'+item.index\"\n           data-toggle=\"tab\"\n           (click)=\"toggle(item.index)\"\n           role=\"tab\" aria-controls=\"home\"\n           [attr.aria-selected]=\"item.isOpen\"\n           [ngClass]=\"{'active':item.isOpen}\">\n          {{item.headerText}}</a>\n      </li>\n    </ul>\n  </li>\n</ul>\n<div class=\"tab-content\" id=\"myTabContent\">\n  <ng-content></ng-content>\n</div>\n",
                        styles: [""]
                    },] },
        ];
        TabComponent.ctorParameters = function () { return []; };
        TabComponent.propDecorators = {
            optionTabAfter: [{ type: i0.Input }],
            tabPanels: [{ type: i0.ContentChildren, args: [TabPanelComponent,] }]
        };
        return TabComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var TabModule = (function () {
        function TabModule() {
        }
        TabModule.decorators = [
            { type: i0.NgModule, args: [{
                        imports: [
                            common.CommonModule,
                            DropdownModule
                        ],
                        declarations: [TabComponent, TabPanelComponent],
                        exports: [TabComponent, TabPanelComponent]
                    },] },
        ];
        return TabModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var TooltipComponent = (function () {
        function TooltipComponent() {
        }
        /**
         * @return {?}
         */
        TooltipComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
            };
        TooltipComponent.decorators = [
            { type: i0.Component, args: [{
                        selector: 'ui-tooltip',
                        template: "<p>\n  tooltip works!\n</p>\n",
                        styles: [""]
                    },] },
        ];
        TooltipComponent.ctorParameters = function () { return []; };
        return TooltipComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var TooltipModule = (function () {
        function TooltipModule() {
        }
        TooltipModule.decorators = [
            { type: i0.NgModule, args: [{
                        imports: [
                            common.CommonModule
                        ],
                        declarations: [TooltipComponent],
                        exports: [TooltipComponent]
                    },] },
        ];
        return TooltipModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */

    exports.NgUikitModule = NgUikitModule;
    exports.AccordionModule = AccordionModule;
    exports.AutocompleteModule = AutocompleteModule;
    exports.CardModule = CardModule;
    exports.CarouselModule = CarouselModule;
    exports.ChartModule = ChartModule;
    exports.DatatableModule = DatatableModule;
    exports.DatepickerModule = DatepickerModule;
    exports.DropdownModule = DropdownModule;
    exports.ModalModule = ModalModule;
    exports.TabModule = TabModule;
    exports.TooltipModule = TooltipModule;
    exports.d = AccordionPanelHeaderComponent;
    exports.c = AccordionPanelComponent;
    exports.b = AccordionComponent;
    exports.e = AutocompleteComponent;
    exports.f = CardComponent;
    exports.g = CarouselComponent;
    exports.h = ChartComponent;
    exports.j = DatatableComponent;
    exports.k = PaginationComponent;
    exports.i = DatatableService;
    exports.l = PagingService;
    exports.n = DatepickerComponent;
    exports.o = DatepickerService;
    exports.m = DropdownComponent;
    exports.p = ModalComponent;
    exports.a = NgUikitComponent;
    exports.r = TabPanelComponent;
    exports.q = TabComponent;
    exports.s = TooltipComponent;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
