// ng-uikit/esm5/lib/autocomplete/autocomplete.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, EventEmitter, Input, Output } from '@angular/core';
import { HttpClient } from '@angular/common/http';
var AutocompleteComponent = /** @class */ (function () {
    function AutocompleteComponent(http) {
        this.http = http;
        this.label = 'Name';
        this.selected = new EventEmitter();
        this.helpText = 'Enter at least 2 char to search';
        this.filteredList = [];
        this.selectedItem = (this.niddle) ? {} : "";
    }
    /**
     * @return {?}
     */
    AutocompleteComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        if (this.sourceURL) {
            this.loadResponse();
        }
    };
    /**
     * @param {?} key
     * @return {?}
     */
    AutocompleteComponent.prototype.filter = /**
     * @param {?} key
     * @return {?}
     */
    function (key) {
        if (key && key.length >= 2) {
            this.filteredList = this.filterResult(key);
            console.log(this.filteredList);
        }
        else {
            this.filteredList = [];
            this.selectedItem = (this.niddle) ? {} : "";
        }
    };
    /**
     * @param {?} key
     * @return {?}
     */
    AutocompleteComponent.prototype.filterResult = /**
     * @param {?} key
     * @return {?}
     */
    function (key) {
        var _this = this;
        /** @type {?} */
        var result = [];
        this.source.forEach(function (item) {
            /** @type {?} */
            var name = (_this.niddle) ? item[_this.niddle].toLowerCase() : item.toString().toLowerCase();
            if (name.indexOf(key.toLowerCase()) !== -1 && (result.length < 10)) {
                result.push(item);
            }
            else {
                return true;
            }
        });
        return result;
    };
    /**
     * @return {?}
     */
    AutocompleteComponent.prototype.loadResponse = /**
     * @return {?}
     */
    function () {
        var _this = this;
        /** @type {?} */
        var url = this.sourceURL || '../../assets/autocomplete.json';
        this.http.get(url).subscribe(function (res) {
            if (Array.isArray(res)) {
                _this.source = res;
            }
        });
    };
    /**
     * @param {?} item
     * @return {?}
     */
    AutocompleteComponent.prototype.getText = /**
     * @param {?} item
     * @return {?}
     */
    function (item) {
        return (this.niddle) ? item[this.niddle] : item;
    };
    /**
     * @param {?} item
     * @return {?}
     */
    AutocompleteComponent.prototype.selectItem = /**
     * @param {?} item
     * @return {?}
     */
    function (item) {
        this.selectedItem = item;
        this.filteredList = [];
        this.selected.emit(item);
    };
    AutocompleteComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ui-autocomplete',
                    template: "<div class=\"form-group ui-autocomplete\">\n  <label for=\"UIAutocomplete\">{{label}}</label>\n  <input *ngIf=\"niddle\" type=\"text\" class=\"form-control\" id=\"UIAutocomplete1\" aria-describedby=\"emailHelp\" placeholder=\"Enter {{label}}\"\n         [ngModel]=\"selectedItem[niddle]\" (keyup)=\"filter($event.target.value)\" autocomplete=\"off\">\n\n  <input *ngIf=\"!niddle\" type=\"text\" class=\"form-control\" id=\"UIAutocomplete\" aria-describedby=\"emailHelp\" placeholder=\"Enter {{label}}\"\n         [ngModel]=\"selectedItem\" (keyup)=\"filter($event.target.value)\" autocomplete=\"off\">\n  <small id=\"emailHelp\" class=\"form-text text-muted\" *ngIf=\"helpText\">{{helpText}}</small>\n  <div class=\"ui-autocomplete-container\">\n    <ul class=\"list-group ui-autocomplete-list\" *ngIf=\"filteredList.length\">\n      <li class=\"list-group-item ui-autocomplete-list-item\" (click)=\"selectItem(item)\" *ngFor=\"let item of filteredList\">{{getText(item)}}</li>\n    </ul>\n  </div>\n</div>\n",
                    styles: [".ui-autocomplete-container{position:relative}.ui-autocomplete-list{position:absolute;z-index:9;width:100%}.ui-autocomplete-list-item:hover{cursor:pointer;color:#fff;background-color:#007bff;border-color:#007bff}"]
                },] },
    ];
    AutocompleteComponent.ctorParameters = function () { return [
        { type: HttpClient }
    ]; };
    AutocompleteComponent.propDecorators = {
        source: [{ type: Input }],
        sourceURL: [{ type: Input }],
        niddle: [{ type: Input }],
        label: [{ type: Input }],
        selected: [{ type: Output }],
        helpText: [{ type: Input }]
    };
    return AutocompleteComponent;
}());
export { AutocompleteComponent };
if (false) {
    /** @type {?} */
    AutocompleteComponent.prototype.source;
    /** @type {?} */
    AutocompleteComponent.prototype.sourceURL;
    /** @type {?} */
    AutocompleteComponent.prototype.niddle;
    /** @type {?} */
    AutocompleteComponent.prototype.label;
    /** @type {?} */
    AutocompleteComponent.prototype.selected;
    /** @type {?} */
    AutocompleteComponent.prototype.helpText;
    /** @type {?} */
    AutocompleteComponent.prototype.filteredList;
    /** @type {?} */
    AutocompleteComponent.prototype.selectedItem;
    /**
     * @type {?}
     * @private
     */
    AutocompleteComponent.prototype.http;
}
