// ng-uikit/esm5/lib/modal/modal.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component } from '@angular/core';
var ModalComponent = /** @class */ (function () {
    function ModalComponent() {
    }
    /**
     * @return {?}
     */
    ModalComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    /**
     * @return {?}
     */
    ModalComponent.prototype.toggle = /**
     * @return {?}
     */
    function () {
        this.showModel = !this.showModel;
    };
    ModalComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ui-modal',
                    template: "<!-- Button trigger modal -->\n<button type=\"button\" class=\"btn btn-primary\" data-toggle=\"modal\" data-target=\"#exampleModal\" (click)=\"toggle()\">\n  Launch demo modal\n</button>\n\n<!-- Modal -->\n<div class=\"modal fade\" id=\"exampleModal\" tabindex=\"-1\" role=\"dialog\" aria-labelledby=\"exampleModalLabel\" aria-hidden=\"true\" [ngClass]=\"{'show':showModel}\">\n  <div class=\"modal-dialog\" role=\"document\">\n    <div class=\"modal-content\">\n      <div class=\"modal-header\">\n        <h5 class=\"modal-title\" id=\"exampleModalLabel\">Modal title</h5>\n        <button type=\"button\" class=\"close\" data-dismiss=\"modal\" aria-label=\"Close\" (click)=\"toggle()\">\n          <span aria-hidden=\"true\">&times;</span>\n        </button>\n      </div>\n      <div class=\"modal-body\">\n        <ng-content></ng-content>\n      </div>\n      <div class=\"modal-footer\">\n        <button type=\"button\" class=\"btn btn-secondary\" data-dismiss=\"modal\" (click)=\"toggle()\">Close</button>\n        <button type=\"button\" class=\"btn btn-primary\" (click)=\"toggle()\">Save changes</button>\n      </div>\n    </div>\n  </div>\n</div>\n<div class=\"modal-backdrop fade show\" *ngIf=\"showModel\"></div>\n",
                    styles: [".show{display:block;padding-right:16px}"]
                },] },
    ];
    ModalComponent.ctorParameters = function () { return []; };
    return ModalComponent;
}());
export { ModalComponent };
if (false) {
    /** @type {?} */
    ModalComponent.prototype.showModel;
}
