// ng-uikit/esm5/lib/card/card.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
/**
 * @record
 */
export function Config() { }
if (false) {
    /** @type {?|undefined} */
    Config.prototype.src;
    /** @type {?|undefined} */
    Config.prototype.alt;
    /** @type {?|undefined} */
    Config.prototype.title;
}
var CardComponent = /** @class */ (function () {
    function CardComponent() {
    }
    /**
     * @return {?}
     */
    CardComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    CardComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ui-card',
                    template: "<div class=\"card\" style=\"width: 18rem;\" *ngIf=\"config\">\n  <img class=\"card-img-top\" src=\"{{config.src}}\" alt=\"{{config.alt || config.title}}\" *ngIf=\"config.src\">\n  <div class=\"card-body\">\n    <h5 class=\"card-title\">{{config.title}}</h5>\n    <ng-content></ng-content>\n  </div>\n</div>\n",
                    styles: [""]
                },] },
    ];
    CardComponent.ctorParameters = function () { return []; };
    CardComponent.propDecorators = {
        config: [{ type: Input }]
    };
    return CardComponent;
}());
export { CardComponent };
if (false) {
    /** @type {?} */
    CardComponent.prototype.config;
}
