// ng-uikit/esm5/lib/datatable/service/datatable.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Injectable } from '@angular/core';
import * as i0 from "@angular/core";
var DatatableService = /** @class */ (function () {
    function DatatableService() {
    }
    /**
     * @param {?} rows
     * @param {?} size
     * @param {?} pageNum
     * @return {?}
     */
    DatatableService.prototype.getPageData = /**
     * @param {?} rows
     * @param {?} size
     * @param {?} pageNum
     * @return {?}
     */
    function (rows, size, pageNum) {
        /** @type {?} */
        var start = (pageNum - 1) * size;
        /** @type {?} */
        var last = (start + size) > rows.length ? rows.length : start + size;
        return {
            startIndex: start,
            rows: rows.slice(start, last),
            lastIndex: last
        };
    };
    DatatableService.decorators = [
        { type: Injectable, args: [{
                    providedIn: 'root'
                },] },
    ];
    DatatableService.ctorParameters = function () { return []; };
    /** @nocollapse */ DatatableService.ngInjectableDef = i0.defineInjectable({ factory: function DatatableService_Factory() { return new DatatableService(); }, token: DatatableService, providedIn: "root" });
    return DatatableService;
}());
export { DatatableService };
