// ng-uikit/esm2015/lib/card/card.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
/**
 * @record
 */
export function Config() { }
if (false) {
    /** @type {?|undefined} */
    Config.prototype.src;
    /** @type {?|undefined} */
    Config.prototype.alt;
    /** @type {?|undefined} */
    Config.prototype.title;
}
export class CardComponent {
    constructor() {
    }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
CardComponent.decorators = [
    { type: Component, args: [{
                selector: 'ui-card',
                template: `<div class="card" style="width: 18rem;" *ngIf="config">
  <img class="card-img-top" src="{{config.src}}" alt="{{config.alt || config.title}}" *ngIf="config.src">
  <div class="card-body">
    <h5 class="card-title">{{config.title}}</h5>
    <ng-content></ng-content>
  </div>
</div>
`,
                styles: [``]
            },] },
];
CardComponent.ctorParameters = () => [];
CardComponent.propDecorators = {
    config: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    CardComponent.prototype.config;
}
