// ng-uikit/esm5/lib/accordion/accordion-panel/accordion-panel.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, ContentChildren, Input, QueryList } from '@angular/core';
import { AccordionPanelHeaderComponent } from "../accordion-panel-header/accordion-panel-header.component";
var AccordionPanelComponent = /** @class */ (function () {
    function AccordionPanelComponent() {
    }
    Object.defineProperty(AccordionPanelComponent.prototype, "hasHeader", {
        get: /**
         * @return {?}
         */
        function () {
            return this.headerComponent && this.headerComponent.length > 0;
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    AccordionPanelComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        this.index = Math.ceil(1000 * Math.random());
    };
    /**
     * @param {?} i
     * @return {?}
     */
    AccordionPanelComponent.prototype.toggle = /**
     * @param {?} i
     * @return {?}
     */
    function (i) {
        this.isOpen = !this.isOpen;
    };
    AccordionPanelComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ui-accordion-panel',
                    template: "<div class=\"card\">\n  <div class=\"card-header\" [attr.id]=\"'heading'+index\">\n    <h5 class=\"mb-0\">\n      <a     data-toggle=\"collapse\" [attr.data-target]=\"'#collapse'+index\"\n              [attr.aria-expanded]=\"isOpen\" [attr.aria-controls]=\"'collapse'+index\" (click)=\"toggle(index)\">\n        <span class=\"ui-accordion-header-text\" *ngIf=\"!hasHeader\">\n          {{header}}\n        </span>\n        <ng-content select=\"ui-accordion-panel-header\" *ngIf=\"hasHeader\"></ng-content>\n      </a>\n    </h5>\n  </div>\n  <div [attr.id]=\"'collapse'+index\" class=\"collapse \" [ngClass]=\"{'show':isOpen}\"\n       [attr.aria-labelledby]=\"'heading'+index\" data-parent=\"#accordionExample\">\n    <div class=\"card-body\">\n      <ng-content></ng-content>\n    </div>\n  </div>\n</div>\n",
                    styles: [""]
                },] },
    ];
    AccordionPanelComponent.ctorParameters = function () { return []; };
    AccordionPanelComponent.propDecorators = {
        header: [{ type: Input }],
        isOpen: [{ type: Input }],
        headerComponent: [{ type: ContentChildren, args: [AccordionPanelHeaderComponent,] }]
    };
    return AccordionPanelComponent;
}());
export { AccordionPanelComponent };
if (false) {
    /** @type {?} */
    AccordionPanelComponent.prototype.header;
    /** @type {?} */
    AccordionPanelComponent.prototype.isOpen;
    /** @type {?} */
    AccordionPanelComponent.prototype.index;
    /** @type {?} */
    AccordionPanelComponent.prototype.headerComponent;
}
