// ng-uikit/esm2015/lib/datatable/service/datatable.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Injectable } from '@angular/core';
import * as i0 from "@angular/core";
export class DatatableService {
    constructor() { }
    /**
     * @param {?} rows
     * @param {?} size
     * @param {?} pageNum
     * @return {?}
     */
    getPageData(rows, size, pageNum) {
        /** @type {?} */
        const start = (pageNum - 1) * size;
        /** @type {?} */
        const last = (start + size) > rows.length ? rows.length : start + size;
        return {
            startIndex: start,
            rows: rows.slice(start, last),
            lastIndex: last
        };
    }
}
DatatableService.decorators = [
    { type: Injectable, args: [{
                providedIn: 'root'
            },] },
];
DatatableService.ctorParameters = () => [];
/** @nocollapse */ DatatableService.ngInjectableDef = i0.defineInjectable({ factory: function DatatableService_Factory() { return new DatatableService(); }, token: DatatableService, providedIn: "root" });
