// ng-uikit/esm2015/lib/datatable/service/paging.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Injectable } from '@angular/core';
import * as i0 from "@angular/core";
export class PagingService {
    constructor() {
    }
    /**
     * @return {?}
     */
    get size() {
        return this._size;
    }
    /**
     * @param {?} value
     * @return {?}
     */
    set size(value) {
        this._size = value;
    }
    /**
     * @return {?}
     */
    get totalCount() {
        return this._totalCount;
    }
    /**
     * @param {?} value
     * @return {?}
     */
    set totalCount(value) {
        this._totalCount = value;
    }
    /**
     * @param {?} totalCount
     * @return {?}
     */
    getListOfPage(totalCount) {
        return Math.ceil(totalCount / this._size);
    }
}
PagingService.decorators = [
    { type: Injectable, args: [{
                providedIn: 'root'
            },] },
];
PagingService.ctorParameters = () => [];
/** @nocollapse */ PagingService.ngInjectableDef = i0.defineInjectable({ factory: function PagingService_Factory() { return new PagingService(); }, token: PagingService, providedIn: "root" });
if (false) {
    /**
     * @type {?}
     * @private
     */
    PagingService.prototype._size;
    /**
     * @type {?}
     * @private
     */
    PagingService.prototype._totalCount;
}
