// ng-uikit/esm2015/lib/autocomplete/autocomplete.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, EventEmitter, Input, Output } from '@angular/core';
import { HttpClient } from '@angular/common/http';
export class AutocompleteComponent {
    /**
     * @param {?} http
     */
    constructor(http) {
        this.http = http;
        this.label = 'Name';
        this.selected = new EventEmitter();
        this.helpText = 'Enter at least 2 char to search';
        this.filteredList = [];
        this.selectedItem = (this.niddle) ? {} : "";
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        if (this.sourceURL) {
            this.loadResponse();
        }
    }
    /**
     * @param {?} key
     * @return {?}
     */
    filter(key) {
        if (key && key.length >= 2) {
            this.filteredList = this.filterResult(key);
            console.log(this.filteredList);
        }
        else {
            this.filteredList = [];
            this.selectedItem = (this.niddle) ? {} : "";
        }
    }
    /**
     * @param {?} key
     * @return {?}
     */
    filterResult(key) {
        /** @type {?} */
        const result = [];
        this.source.forEach((item) => {
            /** @type {?} */
            const name = (this.niddle) ? item[this.niddle].toLowerCase() : item.toString().toLowerCase();
            if (name.indexOf(key.toLowerCase()) !== -1 && (result.length < 10)) {
                result.push(item);
            }
            else {
                return true;
            }
        });
        return result;
    }
    /**
     * @return {?}
     */
    loadResponse() {
        /** @type {?} */
        const url = this.sourceURL || '../../assets/autocomplete.json';
        this.http.get(url).subscribe((res) => {
            if (Array.isArray(res)) {
                this.source = res;
            }
        });
    }
    /**
     * @param {?} item
     * @return {?}
     */
    getText(item) {
        return (this.niddle) ? item[this.niddle] : item;
    }
    /**
     * @param {?} item
     * @return {?}
     */
    selectItem(item) {
        this.selectedItem = item;
        this.filteredList = [];
        this.selected.emit(item);
    }
}
AutocompleteComponent.decorators = [
    { type: Component, args: [{
                selector: 'ui-autocomplete',
                template: `<div class="form-group ui-autocomplete">
  <label for="UIAutocomplete">{{label}}</label>
  <input *ngIf="niddle" type="text" class="form-control" id="UIAutocomplete1" aria-describedby="emailHelp" placeholder="Enter {{label}}"
         [ngModel]="selectedItem[niddle]" (keyup)="filter($event.target.value)" autocomplete="off">

  <input *ngIf="!niddle" type="text" class="form-control" id="UIAutocomplete" aria-describedby="emailHelp" placeholder="Enter {{label}}"
         [ngModel]="selectedItem" (keyup)="filter($event.target.value)" autocomplete="off">
  <small id="emailHelp" class="form-text text-muted" *ngIf="helpText">{{helpText}}</small>
  <div class="ui-autocomplete-container">
    <ul class="list-group ui-autocomplete-list" *ngIf="filteredList.length">
      <li class="list-group-item ui-autocomplete-list-item" (click)="selectItem(item)" *ngFor="let item of filteredList">{{getText(item)}}</li>
    </ul>
  </div>
</div>
`,
                styles: [`.ui-autocomplete-container{position:relative}.ui-autocomplete-list{position:absolute;z-index:9;width:100%}.ui-autocomplete-list-item:hover{cursor:pointer;color:#fff;background-color:#007bff;border-color:#007bff}`]
            },] },
];
AutocompleteComponent.ctorParameters = () => [
    { type: HttpClient }
];
AutocompleteComponent.propDecorators = {
    source: [{ type: Input }],
    sourceURL: [{ type: Input }],
    niddle: [{ type: Input }],
    label: [{ type: Input }],
    selected: [{ type: Output }],
    helpText: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    AutocompleteComponent.prototype.source;
    /** @type {?} */
    AutocompleteComponent.prototype.sourceURL;
    /** @type {?} */
    AutocompleteComponent.prototype.niddle;
    /** @type {?} */
    AutocompleteComponent.prototype.label;
    /** @type {?} */
    AutocompleteComponent.prototype.selected;
    /** @type {?} */
    AutocompleteComponent.prototype.helpText;
    /** @type {?} */
    AutocompleteComponent.prototype.filteredList;
    /** @type {?} */
    AutocompleteComponent.prototype.selectedItem;
    /**
     * @type {?}
     * @private
     */
    AutocompleteComponent.prototype.http;
}
