// ng-uikit/esm5/lib/tab/tab-panel/tab-panel.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
var TabPanelComponent = /** @class */ (function () {
    function TabPanelComponent() {
    }
    /**
     * @return {?}
     */
    TabPanelComponent.prototype.ngOnChanges = /**
     * @return {?}
     */
    function () {
        if (this.isOpen) {
            console.log(this.isOpen);
        }
    };
    /**
     * @return {?}
     */
    TabPanelComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        this.index = Math.ceil(1000 * Math.random());
    };
    TabPanelComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ui-tab-panel',
                    template: "<div class=\"tab-pane\" [attr.id]=\"'tab'+index\" [ngClass]=\"{'active show':isOpen}\" role=\"tabpanel\"\n     aria-labelledby=\"home-tab\">\n  <ng-content></ng-content>\n</div>\n",
                    styles: [".tab-pane{display:none}.tab-pane.active{display:block}"]
                },] },
    ];
    TabPanelComponent.ctorParameters = function () { return []; };
    TabPanelComponent.propDecorators = {
        header: [{ type: Input }],
        isOpen: [{ type: Input }]
    };
    return TabPanelComponent;
}());
export { TabPanelComponent };
if (false) {
    /** @type {?} */
    TabPanelComponent.prototype.header;
    /** @type {?} */
    TabPanelComponent.prototype.isOpen;
    /** @type {?} */
    TabPanelComponent.prototype.index;
}
