// ng-uikit/esm2015/lib/tab/tab-panel/tab-panel.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
export class TabPanelComponent {
    constructor() {
    }
    /**
     * @return {?}
     */
    ngOnChanges() {
        if (this.isOpen) {
            console.log(this.isOpen);
        }
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        this.index = Math.ceil(1000 * Math.random());
    }
}
TabPanelComponent.decorators = [
    { type: Component, args: [{
                selector: 'ui-tab-panel',
                template: `<div class="tab-pane" [attr.id]="'tab'+index" [ngClass]="{'active show':isOpen}" role="tabpanel"
     aria-labelledby="home-tab">
  <ng-content></ng-content>
</div>
`,
                styles: [`.tab-pane{display:none}.tab-pane.active{display:block}`]
            },] },
];
TabPanelComponent.ctorParameters = () => [];
TabPanelComponent.propDecorators = {
    header: [{ type: Input }],
    isOpen: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    TabPanelComponent.prototype.header;
    /** @type {?} */
    TabPanelComponent.prototype.isOpen;
    /** @type {?} */
    TabPanelComponent.prototype.index;
}
