// ng-uikit/esm5/lib/datatable/pagination/pagination.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, EventEmitter, Input, Output } from '@angular/core';
import { PagingService } from '../service/paging.service';
var PaginationComponent = /** @class */ (function () {
    function PaginationComponent(_pgs) {
        this._pgs = _pgs;
        this.size = 10;
        this.pageChange = new EventEmitter();
        this.pageNum = 1;
        this.displayNumbers = 5;
        this.lastPage = 5;
    }
    /**
     * @return {?}
     */
    PaginationComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        this._pgs.size = this.size;
        this._pgs.totalCount = this.totalCount;
        this.pageCount = this._pgs.getListOfPage(this.totalCount);
        this.totalPages = Array(this.pageCount).fill(0).map(function (e, i) { return i + 1; });
        this.pages = this.totalPages.slice(0, 5);
        this.lastPage = this.pages[this.displayNumbers - 1];
    };
    /**
     * @param {?} e
     * @return {?}
     */
    PaginationComponent.prototype.onPageChange = /**
     * @param {?} e
     * @return {?}
     */
    function (e) {
        console.log(e);
        this.pageNum = e;
        this.pageChange.emit({
            pageNum: e
        });
    };
    /**
     * @param {?=} isReverse
     * @return {?}
     */
    PaginationComponent.prototype.shiftBoundary = /**
     * @param {?=} isReverse
     * @return {?}
     */
    function (isReverse) {
        if (isReverse === void 0) { isReverse = false; }
        /** @type {?} */
        var startIndex = (isReverse) ? this.pageNum - this.displayNumbers : this.pageNum - 1;
        /** @type {?} */
        var lastIndex = (isReverse) ? this.pageNum : this.pageNum + this.displayNumbers - 1;
        this.pages = this.totalPages.slice(startIndex, lastIndex);
        this.lastPage = lastIndex;
    };
    /**
     * @return {?}
     */
    PaginationComponent.prototype.next = /**
     * @return {?}
     */
    function () {
        this.pageNum++;
        if (this.pageNum > this.lastPage)
            this.shiftBoundary();
    };
    /**
     * @return {?}
     */
    PaginationComponent.prototype.prev = /**
     * @return {?}
     */
    function () {
        this.pageNum--;
        if ((this.pageNum) % this.displayNumbers === 0) {
            this.shiftBoundary(true);
        }
    };
    PaginationComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ui-pagination',
                    template: "<div class=\"m-3\">\n  <nav aria-label=\"Page navigation example\">\n    <ul class=\"pagination justify-content-center\">\n      <ng-container>\n\n        <li class=\"page-item \" [ngClass]=\"{'disabled':pageNum === 1}\" (click)=\"prev()\" [attr.disabled]=\"pageNum === 1\">\n          <a class=\"page-link\" tabindex=\"-1\">Previous </a>\n        </li>\n\n        <li class=\"page-item\" (click)=\"onPageChange(page)\" *ngFor=\"let page of pages; index as i; first as isFirst; last as isLast;\">\n          <a class=\"page-link\" [ngClass]=\"{'active disabled':pageNum===page}\">{{page}}</a>\n        </li>\n\n        <li class=\"page-item\" *ngIf=\" pageCount > displayNumbers && lastPage < pageCount\" [ngClass]=\"{'disabled':pageNum===pages.length}\" (click)=\"shiftBoundary()\">\n          <a class=\"page-link\">...</a>\n        </li>\n\n        <li class=\"page-item\" *ngIf=\"pageCount > displayNumbers && lastPage < pageCount\" [ngClass]=\"{'disabled':pageNum===pageCount}\" (click)=\"onPageChange(pageCount)\">\n          <a class=\"page-link\"> {{pageCount}}</a>\n        </li>\n\n        <li class=\"page-item\" [ngClass]=\"{'disabled':pageNum === pageCount}\" [attr.disabled]=\"(pageNum === pageCount)?'disabled':''\" (click)=\"next()\">\n          <a class=\"page-link\"> Next</a>\n        </li>\n      </ng-container>\n    </ul>\n  </nav>\n</div>\n\n",
                    styles: [".page-link:not([href]):not([tabindex]){position:relative;display:block;padding:.5rem .75rem;margin-left:-1px;line-height:1.25;color:#337ab7;background-color:#fff;border:1px solid #dee2e6}.page-link.active:not([href]):not([tabindex]){position:relative;display:block;padding:.5rem .75rem;margin-left:-1px;line-height:1.25;color:#fff;background-color:#337ab7;border:1px solid #337ab7}.disabled{pointer-events:none}"]
                },] },
    ];
    PaginationComponent.ctorParameters = function () { return [
        { type: PagingService }
    ]; };
    PaginationComponent.propDecorators = {
        totalCount: [{ type: Input }],
        size: [{ type: Input }],
        pageChange: [{ type: Output }]
    };
    return PaginationComponent;
}());
export { PaginationComponent };
if (false) {
    /** @type {?} */
    PaginationComponent.prototype.totalCount;
    /** @type {?} */
    PaginationComponent.prototype.size;
    /** @type {?} */
    PaginationComponent.prototype.pageChange;
    /** @type {?} */
    PaginationComponent.prototype.pageCount;
    /** @type {?} */
    PaginationComponent.prototype.pages;
    /** @type {?} */
    PaginationComponent.prototype.pageNum;
    /** @type {?} */
    PaginationComponent.prototype.totalPages;
    /** @type {?} */
    PaginationComponent.prototype.displayNumbers;
    /** @type {?} */
    PaginationComponent.prototype.lastPage;
    /**
     * @type {?}
     * @private
     */
    PaginationComponent.prototype._pgs;
}
