// ng-uikit/esm5/lib/tab/tab.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, ContentChildren, Input, QueryList } from '@angular/core';
import { TabPanelComponent } from "./tab-panel/tab-panel.component";
var TabComponent = /** @class */ (function () {
    function TabComponent() {
        this.optionTabAfter = 3;
        this.isShowDropdown = false;
    }
    /**
     * @return {?}
     */
    TabComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    /**
     * @return {?}
     */
    TabComponent.prototype.toggleDropdown = /**
     * @return {?}
     */
    function () {
        this.isShowDropdown = !this.isShowDropdown;
    };
    /**
     * @param {?} i
     * @return {?}
     */
    TabComponent.prototype.toggle = /**
     * @param {?} i
     * @return {?}
     */
    function (i) {
        this.tabPanels.forEach(function (tab) {
            tab.isOpen = (tab.index === i);
        });
    };
    /**
     * @return {?}
     */
    TabComponent.prototype.ngAfterContentInit = /**
     * @return {?}
     */
    function () {
        var _this = this;
        this.tabs = [];
        this.optionTabs = [];
        /** @type {?} */
        var count = 0;
        this.tabPanels.map(function (tab) {
            /** @type {?} */
            var tabItem = {
                headerText: tab.header,
                index: tab.index,
                isOpen: !!tab
            };
            count++;
            _this.optionTabAfter = (_this.optionTabAfter >= _this.tabPanels.length) ? _this.tabPanels.length : _this.optionTabAfter;
            if (count <= _this.optionTabAfter) {
                _this.tabs.push(tabItem);
            }
            else {
                _this.optionTabs.push(tabItem);
            }
        });
        console.log(this.tabs, this.optionTabs);
    };
    TabComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ui-tab',
                    template: "<ul class=\"nav nav-tabs\" id=\"myTab\" role=\"tablist\">\n  <li class=\"nav-item\" *ngFor=\"let item of tabs;index as i\">\n    <a class=\"nav-link \"\n       [attr.id]=\"'tab'+item.index\"\n       data-toggle=\"tab\"\n       (click)=\"toggle(item.index)\"\n       role=\"tab\" aria-controls=\"home\"\n       [attr.aria-selected]=\"item.isOpen\"\n       [ngClass]=\"{'active':item.isOpen}\">\n      {{item.headerText}}\n    </a>\n  </li>\n  <li role=\"presentation\" class=\"dropdown\" *ngIf=\"optionTabs.length\" [ngClass]=\"{'show': isShowDropdown}\">\n    <a class=\"dropdown-toggle\" id=\"myTabDrop\" (click)=\"toggleDropdown()\" data-toggle=\"dropdown\" aria-controls=\"myTabDrop-contents\" attr.aria-expanded=\"{{isShowDropdown}}\">more\n      <span class=\"caret\"></span>\n    </a>\n    <ul class=\"dropdown-menu\" aria-labelledby=\"myTabDrop\" id=\"myTabDrop1-contents\" [ngClass]=\"{'show': isShowDropdown}\">\n      <li *ngFor=\"let item of optionTabs\">\n        <a [attr.id]=\"'tab'+item.index\"\n           data-toggle=\"tab\"\n           (click)=\"toggle(item.index)\"\n           role=\"tab\" aria-controls=\"home\"\n           [attr.aria-selected]=\"item.isOpen\"\n           [ngClass]=\"{'active':item.isOpen}\">\n          {{item.headerText}}</a>\n      </li>\n    </ul>\n  </li>\n</ul>\n<div class=\"tab-content\" id=\"myTabContent\">\n  <ng-content></ng-content>\n</div>\n",
                    styles: [""]
                },] },
    ];
    TabComponent.ctorParameters = function () { return []; };
    TabComponent.propDecorators = {
        optionTabAfter: [{ type: Input }],
        tabPanels: [{ type: ContentChildren, args: [TabPanelComponent,] }]
    };
    return TabComponent;
}());
export { TabComponent };
if (false) {
    /** @type {?} */
    TabComponent.prototype.optionTabAfter;
    /** @type {?} */
    TabComponent.prototype.tabs;
    /** @type {?} */
    TabComponent.prototype.optionTabs;
    /** @type {?} */
    TabComponent.prototype.tabPanels;
    /** @type {?} */
    TabComponent.prototype.isShowDropdown;
}
