// ng-uikit/esm5/lib/dropdown/dropdown.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
/**
 * @record
 */
export function Item() { }
if (false) {
    /** @type {?} */
    Item.prototype.url;
    /** @type {?} */
    Item.prototype.text;
    /** @type {?|undefined} */
    Item.prototype.isDivider;
}
var DropdownComponent = /** @class */ (function () {
    function DropdownComponent() {
        this.items = [];
        this.buttonText = 'Your Button Text';
        this.isShowDropdown = false;
    }
    /**
     * @return {?}
     */
    DropdownComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        this.isShowDropdown = this.isOpen || false;
    };
    /**
     * @return {?}
     */
    DropdownComponent.prototype.toggle = /**
     * @return {?}
     */
    function () {
        this.isShowDropdown = !this.isShowDropdown;
    };
    DropdownComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ui-dropdown',
                    template: "<div class=\"dropdown show\" [ngClass]=\"{'show': isShowDropdown}\">\n  <a class=\"btn btn-secondary dropdown-toggle\" (click)=\"toggle()\" role=\"button\" id=\"dropdownMenuLink\" data-toggle=\"dropdown\"\n     aria-haspopup=\"true\" attr.aria-expanded=\"{{isShowDropdown}}\">\n    {{buttonText}}\n  </a>\n  <div class=\"dropdown-menu dropdown-menu-right\" [ngClass]=\"{'show': isShowDropdown}\" aria-labelledby=\"dropdownMenuLink\">\n    <ng-container *ngFor=\"let item of items\">\n      <a class=\"dropdown-item\" href=\"{{item.url}}\" >{{item.text}}</a>\n      <div class=\"dropdown-divider\" *ngIf=\"item.isDivider\"></div>\n    </ng-container>\n    <ng-content></ng-content>\n  </div>\n</div>\n",
                    styles: [""]
                },] },
    ];
    DropdownComponent.ctorParameters = function () { return []; };
    DropdownComponent.propDecorators = {
        items: [{ type: Input }],
        buttonText: [{ type: Input }],
        isOpen: [{ type: Input }]
    };
    return DropdownComponent;
}());
export { DropdownComponent };
if (false) {
    /** @type {?} */
    DropdownComponent.prototype.items;
    /** @type {?} */
    DropdownComponent.prototype.buttonText;
    /** @type {?} */
    DropdownComponent.prototype.isOpen;
    /** @type {?} */
    DropdownComponent.prototype.isShowDropdown;
}
