// ng-uikit/esm2015/lib/dropdown/dropdown.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
/**
 * @record
 */
export function Item() { }
if (false) {
    /** @type {?} */
    Item.prototype.url;
    /** @type {?} */
    Item.prototype.text;
    /** @type {?|undefined} */
    Item.prototype.isDivider;
}
export class DropdownComponent {
    constructor() {
        this.items = [];
        this.buttonText = 'Your Button Text';
        this.isShowDropdown = false;
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        this.isShowDropdown = this.isOpen || false;
    }
    /**
     * @return {?}
     */
    toggle() {
        this.isShowDropdown = !this.isShowDropdown;
    }
}
DropdownComponent.decorators = [
    { type: Component, args: [{
                selector: 'ui-dropdown',
                template: `<div class="dropdown show" [ngClass]="{'show': isShowDropdown}">
  <a class="btn btn-secondary dropdown-toggle" (click)="toggle()" role="button" id="dropdownMenuLink" data-toggle="dropdown"
     aria-haspopup="true" attr.aria-expanded="{{isShowDropdown}}">
    {{buttonText}}
  </a>
  <div class="dropdown-menu dropdown-menu-right" [ngClass]="{'show': isShowDropdown}" aria-labelledby="dropdownMenuLink">
    <ng-container *ngFor="let item of items">
      <a class="dropdown-item" href="{{item.url}}" >{{item.text}}</a>
      <div class="dropdown-divider" *ngIf="item.isDivider"></div>
    </ng-container>
    <ng-content></ng-content>
  </div>
</div>
`,
                styles: [``]
            },] },
];
DropdownComponent.ctorParameters = () => [];
DropdownComponent.propDecorators = {
    items: [{ type: Input }],
    buttonText: [{ type: Input }],
    isOpen: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    DropdownComponent.prototype.items;
    /** @type {?} */
    DropdownComponent.prototype.buttonText;
    /** @type {?} */
    DropdownComponent.prototype.isOpen;
    /** @type {?} */
    DropdownComponent.prototype.isShowDropdown;
}
