// ng-uikit/esm5/lib/carousel/carousel.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, EventEmitter, Input, Output } from '@angular/core';
var CarouselComponent = /** @class */ (function () {
    function CarouselComponent() {
        this.items = [];
        this.afterNext = new EventEmitter();
        this.afterPrev = new EventEmitter();
        this.width = '600px';
    }
    /**
     * @return {?}
     */
    CarouselComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        this.activeIndex = 0;
        this.count = this.items.length;
    };
    /**
     * @return {?}
     */
    CarouselComponent.prototype.next = /**
     * @return {?}
     */
    function () {
        (this.activeIndex < this.count - 1) ? this.activeIndex++ : this.activeIndex = 0;
        this.afterNext.emit({
            index: this.activeIndex,
            item: this.items[this.activeIndex]
        });
    };
    /**
     * @return {?}
     */
    CarouselComponent.prototype.prev = /**
     * @return {?}
     */
    function () {
        (this.activeIndex >= 1) ? this.activeIndex-- : this.activeIndex = this.count - 1;
        this.afterPrev.emit({
            index: this.activeIndex,
            item: this.items[this.activeIndex]
        });
    };
    /**
     * @param {?} i
     * @return {?}
     */
    CarouselComponent.prototype.goTo = /**
     * @param {?} i
     * @return {?}
     */
    function (i) {
        this.activeIndex = i;
    };
    CarouselComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ui-carousel',
                    template: "<div id=\"carouselExampleIndicators\" class=\"carousel slide\" data-ride=\"carousel\" [ngStyle]=\"{width: width}\">\n  <ol class=\"carousel-indicators\">\n      <li data-target=\"#carouselExampleIndicators\" attr.data-slide-to=\"{{i}}\" *ngFor=\"let item of items; let i=index;\"\n          [ngClass]=\"{'active':i == activeIndex}\" (click)=\"goTo(i)\">\n      </li>\n  </ol>\n  <div class=\"carousel-inner\">\n    <div class=\"carousel-item\" *ngFor=\"let item of items; let i=index;\" [ngClass]=\"{'active':i == activeIndex}\">\n      <img src=\"{{item.src}}\" alt=\"{{item.alt}}\">\n      <div class=\"carousel-caption d-none d-md-block\">\n        <h5>{{item.alt}}</h5>\n        <p>{{item.alt}}</p>\n      </div>\n    </div>\n  </div>\n  <a class=\"carousel-control-prev\" (click)=\"prev()\" role=\"button\" data-slide=\"prev\">\n    <span class=\"carousel-control-prev-icon\" aria-hidden=\"true\"></span>\n    <span class=\"sr-only\">Previous</span>\n  </a>\n  <a class=\"carousel-control-next\" (click)=\"next()\" role=\"button\" data-slide=\"next\">\n    <span class=\"carousel-control-next-icon\" aria-hidden=\"true\"></span>\n    <span class=\"sr-only\">Next</span>\n  </a>\n</div>\n",
                    styles: [""]
                },] },
    ];
    CarouselComponent.ctorParameters = function () { return []; };
    CarouselComponent.propDecorators = {
        items: [{ type: Input }],
        afterNext: [{ type: Output }],
        afterPrev: [{ type: Output }]
    };
    return CarouselComponent;
}());
export { CarouselComponent };
if (false) {
    /** @type {?} */
    CarouselComponent.prototype.items;
    /** @type {?} */
    CarouselComponent.prototype.afterNext;
    /** @type {?} */
    CarouselComponent.prototype.afterPrev;
    /** @type {?} */
    CarouselComponent.prototype.activeIndex;
    /** @type {?} */
    CarouselComponent.prototype.count;
    /** @type {?} */
    CarouselComponent.prototype.width;
}
