// ng-uikit/esm2015/lib/accordion/accordion-panel/accordion-panel.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, ContentChildren, Input, QueryList } from '@angular/core';
import { AccordionPanelHeaderComponent } from "../accordion-panel-header/accordion-panel-header.component";
export class AccordionPanelComponent {
    constructor() {
    }
    /**
     * @return {?}
     */
    get hasHeader() {
        return this.headerComponent && this.headerComponent.length > 0;
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        this.index = Math.ceil(1000 * Math.random());
    }
    /**
     * @param {?} i
     * @return {?}
     */
    toggle(i) {
        this.isOpen = !this.isOpen;
    }
}
AccordionPanelComponent.decorators = [
    { type: Component, args: [{
                selector: 'ui-accordion-panel',
                template: `<div class="card">
  <div class="card-header" [attr.id]="'heading'+index">
    <h5 class="mb-0">
      <a     data-toggle="collapse" [attr.data-target]="'#collapse'+index"
              [attr.aria-expanded]="isOpen" [attr.aria-controls]="'collapse'+index" (click)="toggle(index)">
        <span class="ui-accordion-header-text" *ngIf="!hasHeader">
          {{header}}
        </span>
        <ng-content select="ui-accordion-panel-header" *ngIf="hasHeader"></ng-content>
      </a>
    </h5>
  </div>
  <div [attr.id]="'collapse'+index" class="collapse " [ngClass]="{'show':isOpen}"
       [attr.aria-labelledby]="'heading'+index" data-parent="#accordionExample">
    <div class="card-body">
      <ng-content></ng-content>
    </div>
  </div>
</div>
`,
                styles: [``]
            },] },
];
AccordionPanelComponent.ctorParameters = () => [];
AccordionPanelComponent.propDecorators = {
    header: [{ type: Input }],
    isOpen: [{ type: Input }],
    headerComponent: [{ type: ContentChildren, args: [AccordionPanelHeaderComponent,] }]
};
if (false) {
    /** @type {?} */
    AccordionPanelComponent.prototype.header;
    /** @type {?} */
    AccordionPanelComponent.prototype.isOpen;
    /** @type {?} */
    AccordionPanelComponent.prototype.index;
    /** @type {?} */
    AccordionPanelComponent.prototype.headerComponent;
}
