// ng-uikit/esm2015/lib/tab/tab.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, ContentChildren, Input, QueryList } from '@angular/core';
import { TabPanelComponent } from "./tab-panel/tab-panel.component";
export class TabComponent {
    constructor() {
        this.optionTabAfter = 3;
        this.isShowDropdown = false;
    }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
    /**
     * @return {?}
     */
    toggleDropdown() {
        this.isShowDropdown = !this.isShowDropdown;
    }
    /**
     * @param {?} i
     * @return {?}
     */
    toggle(i) {
        this.tabPanels.forEach(tab => {
            tab.isOpen = (tab.index === i);
        });
    }
    /**
     * @return {?}
     */
    ngAfterContentInit() {
        this.tabs = [];
        this.optionTabs = [];
        /** @type {?} */
        let count = 0;
        this.tabPanels.map(tab => {
            /** @type {?} */
            const tabItem = {
                headerText: tab.header,
                index: tab.index,
                isOpen: !!tab
            };
            count++;
            this.optionTabAfter = (this.optionTabAfter >= this.tabPanels.length) ? this.tabPanels.length : this.optionTabAfter;
            if (count <= this.optionTabAfter) {
                this.tabs.push(tabItem);
            }
            else {
                this.optionTabs.push(tabItem);
            }
        });
        console.log(this.tabs, this.optionTabs);
    }
}
TabComponent.decorators = [
    { type: Component, args: [{
                selector: 'ui-tab',
                template: `<ul class="nav nav-tabs" id="myTab" role="tablist">
  <li class="nav-item" *ngFor="let item of tabs;index as i">
    <a class="nav-link "
       [attr.id]="'tab'+item.index"
       data-toggle="tab"
       (click)="toggle(item.index)"
       role="tab" aria-controls="home"
       [attr.aria-selected]="item.isOpen"
       [ngClass]="{'active':item.isOpen}">
      {{item.headerText}}
    </a>
  </li>
  <li role="presentation" class="dropdown" *ngIf="optionTabs.length" [ngClass]="{'show': isShowDropdown}">
    <a class="dropdown-toggle" id="myTabDrop" (click)="toggleDropdown()" data-toggle="dropdown" aria-controls="myTabDrop-contents" attr.aria-expanded="{{isShowDropdown}}">more
      <span class="caret"></span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="myTabDrop" id="myTabDrop1-contents" [ngClass]="{'show': isShowDropdown}">
      <li *ngFor="let item of optionTabs">
        <a [attr.id]="'tab'+item.index"
           data-toggle="tab"
           (click)="toggle(item.index)"
           role="tab" aria-controls="home"
           [attr.aria-selected]="item.isOpen"
           [ngClass]="{'active':item.isOpen}">
          {{item.headerText}}</a>
      </li>
    </ul>
  </li>
</ul>
<div class="tab-content" id="myTabContent">
  <ng-content></ng-content>
</div>
`,
                styles: [``]
            },] },
];
TabComponent.ctorParameters = () => [];
TabComponent.propDecorators = {
    optionTabAfter: [{ type: Input }],
    tabPanels: [{ type: ContentChildren, args: [TabPanelComponent,] }]
};
if (false) {
    /** @type {?} */
    TabComponent.prototype.optionTabAfter;
    /** @type {?} */
    TabComponent.prototype.tabs;
    /** @type {?} */
    TabComponent.prototype.optionTabs;
    /** @type {?} */
    TabComponent.prototype.tabPanels;
    /** @type {?} */
    TabComponent.prototype.isShowDropdown;
}
